diff --git a/node_modules/matrix-react-sdk/res/css/views/rooms/_LegacyRoomHeader.pcss b/node_modules/matrix-react-sdk/res/css/views/rooms/_LegacyRoomHeader.pcss
index 17f1dfe..ac831cd 100644
--- a/node_modules/matrix-react-sdk/res/css/views/rooms/_LegacyRoomHeader.pcss
+++ b/node_modules/matrix-react-sdk/res/css/views/rooms/_LegacyRoomHeader.pcss
@@ -135,6 +135,20 @@ limitations under the License.
     cursor: pointer;
 }
 
+/* :tchap: largely inspired from ;mx_RoomHeader_topic but I couldn't get this add-on to work with $variable */
+.tc_RoomHeader_external {
+    color: var(--external-color);
+    width: 80px;
+    font-weight: 400;
+    font-size: 1.3rem;
+    line-height: 1.3rem;
+    max-height: calc(1.3rem * 2);
+    overflow: hidden;
+    -webkit-line-clamp: 2;
+    -webkit-box-orient: vertical;
+    display: -webkit-box;
+}
+
 .mx_LegacyRoomHeader_topic {
     $lines: 2;
 
diff --git a/node_modules/matrix-react-sdk/src/components/views/rooms/LegacyRoomHeader.tsx b/node_modules/matrix-react-sdk/src/components/views/rooms/LegacyRoomHeader.tsx
index 315332f..5d1380f 100644
--- a/node_modules/matrix-react-sdk/src/components/views/rooms/LegacyRoomHeader.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/rooms/LegacyRoomHeader.tsx
@@ -70,6 +70,7 @@ import { Alignment } from "../elements/Tooltip";
 import RoomCallBanner from "../beacon/RoomCallBanner";
 import { shouldShowComponent } from "../../../customisations/helpers/UIComponents";
 import { UIComponent } from "../../../settings/UIFeature";
+import TchapExternalRoomHeader from "../../../../../../src/tchap/components/views/rooms/TchapExternalRoomHeader";
 
 class DisabledWithReason {
     public constructor(public readonly reason: string) {}
@@ -814,6 +815,7 @@ export default class RoomHeader extends React.Component<IProps, IState> {
                     <div className="mx_LegacyRoomHeader_avatar">{roomAvatar}</div>
                     {icon}
                     {name}
+                    <TchapExternalRoomHeader room={this.props.room}></TchapExternalRoomHeader>
                     {searchStatus}
                     {topicElement}
                     {betaPill}
