diff --git a/node_modules/matrix-react-sdk/res/css/views/rooms/_RoomHeader.pcss b/node_modules/matrix-react-sdk/res/css/views/rooms/_RoomHeader.pcss
index 80e3bfd..5c7cb2c 100644
--- a/node_modules/matrix-react-sdk/res/css/views/rooms/_RoomHeader.pcss
+++ b/node_modules/matrix-react-sdk/res/css/views/rooms/_RoomHeader.pcss
@@ -149,6 +149,20 @@ limitations under the License.
     cursor: pointer;
 }
 
+.tc_RoomHeader_external {
+    color: orange;
+    width: 80px;
+    font-weight: 400;
+    font-size: 1.3rem;
+    line-height: 1.3rem;
+    max-height: calc(1.3rem * 2);
+    overflow: hidden;
+    -webkit-line-clamp: 2;
+    -webkit-box-orient: vertical;
+    display: -webkit-box;        
+}
+
+
 .mx_RoomHeader_topic {
     $lineHeight: $font-16px;
     $lines: 2;
diff --git a/node_modules/matrix-react-sdk/src/components/views/rooms/RoomHeader.tsx b/node_modules/matrix-react-sdk/src/components/views/rooms/RoomHeader.tsx
index 0d01e03..50ec61c 100644
--- a/node_modules/matrix-react-sdk/src/components/views/rooms/RoomHeader.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/rooms/RoomHeader.tsx
@@ -65,7 +65,8 @@ import IconizedContextMenu, {
     IconizedContextMenuOptionList,
 } from "../context_menus/IconizedContextMenu";
 import { ViewRoomPayload } from "../../../dispatcher/payloads/ViewRoomPayload";
-
+import TchapExternalRoomHeader from "../../../../../../src/components/views/rooms/TchapExternalRoomHeader";
+import "../../../../../../res/css/views/rooms/_TchapExternalRoomHeader.pcss";
 class DisabledWithReason {
     constructor(public readonly reason: string) { }
 }
@@ -583,6 +584,7 @@ export default class RoomHeader extends React.Component<IProps, IState> {
                     <div className="mx_RoomHeader_avatar">{ roomAvatar }</div>
                     <div className="mx_RoomHeader_e2eIcon">{ e2eIcon }</div>
                     { name }
+                    <TchapExternalRoomHeader room={this.props.room}></TchapExternalRoomHeader>
                     { searchStatus }
                     { topicElement }
                     { betaPill }
