diff --git a/node_modules/matrix-react-sdk/src/components/structures/auth/Login.tsx b/node_modules/matrix-react-sdk/src/components/structures/auth/Login.tsx
index c00aa90..25d6f0d 100644
--- a/node_modules/matrix-react-sdk/src/components/structures/auth/Login.tsx
+++ b/node_modules/matrix-react-sdk/src/components/structures/auth/Login.tsx
@@ -165,7 +165,17 @@ export default class LoginComponent extends React.PureComponent<IProps, IState>
 
     isBusy = () => this.state.busy || this.props.busy;
 
+    fetchHomeserverForEmail = async (email) => {
+        // todo draw a random server
+        // const tchapHostsList = this._shuffle(SdkConfig.get()['hs_url_list']);
+        const serverBaseToQuery = "https://matrix.agent.diplomatie.tchap.gouv.fr";
+        const infoUrl = "/_matrix/identity/api/v1/info?medium=email&address=";
+
+        return fetch(serverBaseToQuery + infoUrl + email).then((response) => response.json());
+    };
+
     onPasswordLogin = async (username, phoneCountry, phoneNumber, password) => {
+        /* :TCHAP: remove alive check, we don't have server yet todo */
         if (!this.state.serverIsAlive) {
             this.setState({ busy: true });
             // Do a quick liveliness check on the URLs
@@ -199,6 +209,40 @@ export default class LoginComponent extends React.PureComponent<IProps, IState>
             loginIncorrect: false,
         });
 
+        /* :TCHAP: fetch homeserver corresponding to email */
+        console.log("onPasswordLogin", username);
+        const serverResult = await this.fetchHomeserverForEmail(username);
+        console.log("onPasswordLogin serverResult", serverResult);
+        // todo deal with error case
+
+        const serverName = "DINUM"; // todo look it up in config
+
+        // Make a ValidatedServerConfig
+        const discoveryResult = {
+            "m.homeserver": {
+                state: "SUCCESS",
+                error: null,
+                base_url: "https://matrix." + serverResult.hs,
+                server_name: serverName,
+            },
+            "m.identity_server": {
+                state: "PROMPT",
+                error: null,
+                base_url: "https://matrix." + serverResult.hs, // On Tchap our Identity server urls and home server urls are the same
+                server_name: serverName,
+            },
+        };
+        const validatedServerConf = AutoDiscoveryUtils.buildValidatedConfigFromDiscovery(
+            discoveryResult['m.homeserver'].server_name, discoveryResult);
+
+        // Simulate the end of the serverPicker component flow.
+        this.props.onServerConfigChange(validatedServerConf);
+
+
+        this.initLoginLogic(validatedServerConf);
+        // todo what happens if same or different to defaultServer ? What is value of defaultServer ?
+        /** end :TCHAP: */
+
         this.loginLogic.loginViaPassword(
             username, phoneCountry, phoneNumber, password,
         ).then((data) => {
@@ -606,10 +650,12 @@ export default class LoginComponent extends React.PureComponent<IProps, IState>
                     </h1>
                     { errorTextSection }
                     { serverDeadSection }
+                    { /* :TCHAP: remove server picker, we're going to query backend to find it
                     <ServerPicker
                         serverConfig={this.props.serverConfig}
                         onServerConfigChange={this.props.onServerConfigChange}
                     />
+                    end :TCHAP :*/}
                     { this.renderLoginComponentForFlows() }
                     { footer }
                 </AuthBody>
