diff --git a/node_modules/matrix-react-sdk/src/components/structures/auth/Login.tsx b/node_modules/matrix-react-sdk/src/components/structures/auth/Login.tsx
index c00aa90..39232b3 100644
--- a/node_modules/matrix-react-sdk/src/components/structures/auth/Login.tsx
+++ b/node_modules/matrix-react-sdk/src/components/structures/auth/Login.tsx
@@ -38,6 +38,7 @@ import AuthBody from "../../views/auth/AuthBody";
 import AuthHeader from "../../views/auth/AuthHeader";
 import AccessibleButton from '../../views/elements/AccessibleButton';
 import { ValidatedServerConfig } from '../../../utils/ValidatedServerConfig';
+import TchapUtils from '../../../../../../src/util/TchapUtils'; // :TCHAP:
 
 // These are used in several places, and come from the js-sdk's autodiscovery
 // stuff. We define them here so that they'll be picked up by i18n.
@@ -165,7 +166,17 @@ export default class LoginComponent extends React.PureComponent<IProps, IState>
 
     isBusy = () => this.state.busy || this.props.busy;
 
+    tchap_setServerInMemory = async (serverConfig) => {
+        const validatedServerConf = TchapUtils.makeValidatedServerConfig(serverConfig);
+
+        // Simulate the end of the serverPicker component flow.
+        this.props.onServerConfigChange(validatedServerConf);
+
+        await this.initLoginLogic(validatedServerConf);
+    };
+
     onPasswordLogin = async (username, phoneCountry, phoneNumber, password) => {
+        /* :TCHAP: remove alive check, we don't know which server to call yet.
         if (!this.state.serverIsAlive) {
             this.setState({ busy: true });
             // Do a quick liveliness check on the URLs
@@ -190,7 +201,7 @@ export default class LoginComponent extends React.PureComponent<IProps, IState>
             if (!aliveAgain) {
                 return;
             }
-        }
+        } end :TCHAP: */
 
         this.setState({
             busy: true,
@@ -199,6 +210,22 @@ export default class LoginComponent extends React.PureComponent<IProps, IState>
             loginIncorrect: false,
         });
 
+        /* :TCHAP: fetch homeserver corresponding to email */
+        const serverResult = await TchapUtils.fetchHomeserverForEmail(username);
+
+        if (!serverResult) {
+            this.setState({
+                busy: false,
+                busyLoggingIn: false,
+                errorText: _t('Server unavailable, overloaded, or something else went wrong.'), // reuse existing string
+                loginIncorrect: false,
+            });
+            return;
+        }
+
+        await this.tchap_setServerInMemory(serverResult);
+        /** end :TCHAP: */
+
         this.loginLogic.loginViaPassword(
             username, phoneCountry, phoneNumber, password,
         ).then((data) => {
@@ -261,6 +288,11 @@ export default class LoginComponent extends React.PureComponent<IProps, IState>
                 } else {
                     errorText = _t('Incorrect username and/or password.');
                 }
+            // :TCHAP: display proper message for TOO_MANY_REQUESTS
+            } else if (error.httpStatus === 429) {
+                console.log('fff error', error, error.httpStatus, error.errcode);
+                errorText = _t("You have made to many requests and have been blocked. Try again later.");
+            // end :TCHAP:
             } else {
                 // other errors, not specific to doing a password login
                 errorText = this.errorTextFromError(error);
@@ -606,10 +638,12 @@ export default class LoginComponent extends React.PureComponent<IProps, IState>
                     </h1>
                     { errorTextSection }
                     { serverDeadSection }
+                    { /* :TCHAP: remove server picker, we don't allow user to chose, server is assigned to each email.
                     <ServerPicker
                         serverConfig={this.props.serverConfig}
                         onServerConfigChange={this.props.onServerConfigChange}
                     />
+                    end :TCHAP :*/}
                     { this.renderLoginComponentForFlows() }
                     { footer }
                 </AuthBody>
diff --git a/node_modules/matrix-react-sdk/src/components/views/auth/PasswordLogin.tsx b/node_modules/matrix-react-sdk/src/components/views/auth/PasswordLogin.tsx
index f6b4e19..f3dba35 100644
--- a/node_modules/matrix-react-sdk/src/components/views/auth/PasswordLogin.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/auth/PasswordLogin.tsx
@@ -80,7 +80,10 @@ export default class PasswordLogin extends React.PureComponent<IProps, IState> {
         this.state = {
             // Field error codes by field ID
             fieldValid: {},
-            loginType: LoginField.MatrixId,
+            // :TCHAP: force email login
+            // loginType: LoginField.MatrixId,
+            loginType: LoginField.Email,
+            //
             password: "",
         };
     }
@@ -417,7 +420,11 @@ export default class PasswordLogin extends React.PureComponent<IProps, IState> {
         return (
             <div>
                 <form onSubmit={this.onSubmitForm}>
-                    { loginType }
+                    {
+                        /* :TCHAP: remove loginType selector, we only want Email loginType
+                        loginType
+                        */
+                    }
                     { loginField }
                     <Field
                         id="mx_LoginForm_password"
