diff --git a/node_modules/matrix-react-sdk/src/IdentityAuthClient.tsx b/node_modules/matrix-react-sdk/src/IdentityAuthClient.tsx
index c27640c..b945877 100644
--- a/node_modules/matrix-react-sdk/src/IdentityAuthClient.tsx
+++ b/node_modules/matrix-react-sdk/src/IdentityAuthClient.tsx
@@ -30,6 +30,7 @@ import {
 } from "./utils/IdentityServerUtils";
 import QuestionDialog from "./components/views/dialogs/QuestionDialog";
 import { abbreviateUrl } from "./utils/UrlUtils";
+import TchapUIFeature from "../../../src/tchap/util/TchapUIFeature";
 
 export class AbortedIdentityActionError extends Error {}
 
@@ -135,6 +136,18 @@ export default class IdentityAuthClient {
             throw e;
         }
 
+        // :TCHAP: no need confirmation of Terms and Conditions to set a default identity server as we trust our backend servers 
+        if (TchapUIFeature.autoAcceptTermsAndConditions){
+            if (
+                !this.tempClient &&
+                !doesAccountDataHaveIdentityServer(this.matrixClient) 
+            ) {
+                setToDefaultIdentityServer(this.matrixClient);
+            }
+            return;
+        }
+        // end :TCHAP:
+
         if (
             !this.tempClient &&
             !doesAccountDataHaveIdentityServer(this.matrixClient) &&
diff --git a/node_modules/matrix-react-sdk/src/Terms.ts b/node_modules/matrix-react-sdk/src/Terms.ts
index d31b635..9b250b1 100644
--- a/node_modules/matrix-react-sdk/src/Terms.ts
+++ b/node_modules/matrix-react-sdk/src/Terms.ts
@@ -21,6 +21,11 @@ import { MatrixClient } from "matrix-js-sdk/src/matrix";
 
 import Modal from "./Modal";
 import TermsDialog from "./components/views/dialogs/TermsDialog";
+import {
+    doesAccountDataHaveIdentityServer,
+    setToDefaultIdentityServer,
+} from './utils/IdentityServerUtils';
+import TchapUIFeature from "../../../src/tchap/util/TchapUIFeature";
 
 export class TermsNotSignedError extends Error {}
 
@@ -80,6 +85,15 @@ export async function startTermsFlow(
     services: Service[],
     interactionCallback: TermsInteractionCallback = dialogTermsInteractionCallback,
 ): Promise<void> {
+    // :TCHAP: no need to go through Terms flow as we trust our backend servers
+    if (TchapUIFeature.autoAcceptTermsAndConditions){
+        if (!doesAccountDataHaveIdentityServer(client)) {
+            setToDefaultIdentityServer(client);
+        }
+        return;
+    }
+    // end :TCHAP:
+
     const termsPromises = services.map((s) => client.getTerms(s.serviceType, s.baseUrl));
 
     /*
