diff --git a/node_modules/matrix-react-sdk/src/components/views/dialogs/LogoutDialog.tsx b/node_modules/matrix-react-sdk/src/components/views/dialogs/LogoutDialog.tsx
index cea5d31..8ed11bd 100644
--- a/node_modules/matrix-react-sdk/src/components/views/dialogs/LogoutDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/dialogs/LogoutDialog.tsx
@@ -157,6 +157,7 @@ export default class LogoutDialog extends React.Component<IProps, IState> {
                     <div className="mx_Dialog_content" id='mx_Dialog_content'>
                         { description }
                     </div>
+                    { /** :TCHAP: do not suggest key backup. Suggest exporting keys as primary method.
                     <DialogButtons primaryButton={setupButtonCaption}
                         hasCancel={false}
                         onPrimaryButtonClick={this.onSetRecoveryMethodClick}
@@ -172,6 +173,17 @@ export default class LogoutDialog extends React.Component<IProps, IState> {
                             { _t("Manually export keys") }
                         </button></p>
                     </details>
+                    */ }
+                    <DialogButtons primaryButton={_t("Manually export keys")}
+                        hasCancel={false}
+                        onPrimaryButtonClick={this.onExportE2eKeysClicked}
+                        focus={true}
+                    >
+                        <button onClick={this.onLogoutConfirm}>
+                            { _t("I don't want my encrypted messages") }
+                        </button>
+                    </DialogButtons>
+                    { /** end :TCHAP: */ }
                 </div>;
             }
             // Not quite a standard question dialog as the primary button cancels
