diff --git a/node_modules/matrix-react-sdk/src/components/views/rooms/MessageComposer.tsx b/node_modules/matrix-react-sdk/src/components/views/rooms/MessageComposer.tsx
index bf5f6b4..170136e 100644
--- a/node_modules/matrix-react-sdk/src/components/views/rooms/MessageComposer.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/rooms/MessageComposer.tsx
@@ -60,6 +60,7 @@ import { setUpVoiceBroadcastPreRecording } from "../../../voice-broadcast/utils/
 import { SdkContextClass } from "../../../contexts/SDKContext";
 import { VoiceBroadcastInfoState } from "../../../voice-broadcast";
 import { createCantStartVoiceMessageBroadcastDialog } from "../dialogs/CantStartVoiceMessageBroadcastDialog";
+import TchapUtils from '../../../../../../src/tchap/util/TchapUtils'; // :TCHAP:
 
 let instanceCount = 0;
 
@@ -614,7 +615,7 @@ export class MessageComposer extends React.Component<IProps, IState> {
                                     relation={this.props.relation}
                                     onRecordStartEndClick={this.onRecordStartEndClick}
                                     setStickerPickerOpen={this.setStickerPickerOpen}
-                                    showLocationButton={!window.electron}
+                                    showLocationButton={!window.electron && TchapUtils.isMapConfigured() } // :TCHAP: showLocationButton only if map server is configured.
                                     showPollsButton={this.state.showPollsButton}
                                     showStickersButton={this.showStickersButton}
                                     isRichTextEnabled={this.state.isRichTextEnabled}
