diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/Notifications.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/Notifications.tsx
index 35c7fa3..efc9972 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/Notifications.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/Notifications.tsx
@@ -672,6 +672,7 @@ export default class Notifications extends React.PureComponent<IProps, IState> {
             return masterSwitch;
         }
 
+        /* :TCHAP disable for now
         const emailSwitches = (this.state.threepids || [])
             .filter((t) => t.medium === ThreepidMedium.Email)
             .map((e) => (
@@ -684,6 +685,7 @@ export default class Notifications extends React.PureComponent<IProps, IState> {
                     disabled={this.state.phase === Phase.Persisting}
                 />
             ));
+        :tchap end */
 
         return (
             <>
@@ -723,7 +725,9 @@ export default class Notifications extends React.PureComponent<IProps, IState> {
                     </>
                 )}
 
-                {emailSwitches}
+                {/* :TCHAP disable for now
+                emailSwitches
+                :tchap end */}
             </>
         );
     }
