diff --git a/node_modules/matrix-react-sdk/src/components/views/dialogs/SpacePreferencesDialog.tsx b/node_modules/matrix-react-sdk/src/components/views/dialogs/SpacePreferencesDialog.tsx
index 3042405..2123c91 100644
--- a/node_modules/matrix-react-sdk/src/components/views/dialogs/SpacePreferencesDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/dialogs/SpacePreferencesDialog.tsx
@@ -53,7 +53,9 @@ const SpacePreferencesAppearanceTab: React.FC<Pick<IProps, "space">> = ({ space
                         );
                     }}
                 >
-                    {_t("People")}
+                    {/* TCHAP: change label {_t("People")} */}
+                    {_t("Direct Messages")}
+                    {/* end TCHAP */}
                 </StyledCheckbox>
                 <p>
                     {_t(
diff --git a/node_modules/matrix-react-sdk/src/components/views/rooms/RoomList.tsx b/node_modules/matrix-react-sdk/src/components/views/rooms/RoomList.tsx
index 98a3260..deed0b6 100644
--- a/node_modules/matrix-react-sdk/src/components/views/rooms/RoomList.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/rooms/RoomList.tsx
@@ -404,7 +404,7 @@ const TAG_AESTHETICS: TagAestheticsMap = {
         defaultHidden: false,
     },
     [DefaultTagID.DM]: {
-        sectionLabel: _td("People"),
+        sectionLabel: _td("Direct Messages"),
         isInvite: false,
         defaultHidden: false,
         AuxButtonComponent: DmAuxButton,
diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SidebarUserSettingsTab.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SidebarUserSettingsTab.tsx
index 1a11e0d..b4800b3 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SidebarUserSettingsTab.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SidebarUserSettingsTab.tsx
@@ -105,7 +105,9 @@ const SidebarUserSettingsTab: React.FC = () => {
                     onChange={onMetaSpaceChangeFactory(MetaSpace.People, "WebSettingsSidebarTabSpacesCheckbox")}
                     className="mx_SidebarUserSettingsTab_peopleCheckbox"
                 >
-                    {_t("People")}
+                    {/* TCHAP: change label {_t("People")} */}
+                    {_t("Direct Messages")}
+                    {/* end TCHAP */}
                 </StyledCheckbox>
                 <div className="mx_SidebarUserSettingsTab_checkboxMicrocopy">
                     {_t("Group all your people in one place.")}
diff --git a/node_modules/matrix-react-sdk/src/components/views/spaces/QuickSettingsButton.tsx b/node_modules/matrix-react-sdk/src/components/views/spaces/QuickSettingsButton.tsx
index 1adcd41..2054ab9 100644
--- a/node_modules/matrix-react-sdk/src/components/views/spaces/QuickSettingsButton.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/spaces/QuickSettingsButton.tsx
@@ -105,7 +105,9 @@ const QuickSettingsButton: React.FC<{
                     onChange={onMetaSpaceChangeFactory(MetaSpace.People, "WebQuickSettingsPinToSidebarCheckbox")}
                 >
                     <MembersIcon className="mx_QuickSettingsButton_icon" />
-                    {_t("People")}
+                    {/* TCHAP: change label {_t("People")} */}
+                    {_t("Direct Messages")}
+                    {/* end TCHAP */}
                 </StyledCheckbox>
                 <AccessibleButton
                     className="mx_QuickSettingsButton_moreOptionsButton"
diff --git a/node_modules/matrix-react-sdk/src/stores/spaces/index.ts b/node_modules/matrix-react-sdk/src/stores/spaces/index.ts
index 0cc4bba..165e2bb 100644
--- a/node_modules/matrix-react-sdk/src/stores/spaces/index.ts
+++ b/node_modules/matrix-react-sdk/src/stores/spaces/index.ts
@@ -42,7 +42,10 @@ export const getMetaSpaceName = (spaceKey: MetaSpace, allRoomsInHome = false): s
         case MetaSpace.Favourites:
             return _t("Favourites");
         case MetaSpace.People:
-            return _t("People");
+            // TCHAP:
+            // return _t("People")
+            return _t("Direct Messages");
+            // end TCHAP
         case MetaSpace.Orphans:
             return _t("Other rooms");
     }
