diff --git a/node_modules/matrix-react-sdk/src/components/views/dialogs/SpacePreferencesDialog.tsx b/node_modules/matrix-react-sdk/src/components/views/dialogs/SpacePreferencesDialog.tsx
index 45f60d1..1844ae7 100644
--- a/node_modules/matrix-react-sdk/src/components/views/dialogs/SpacePreferencesDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/dialogs/SpacePreferencesDialog.tsx
@@ -55,7 +55,7 @@ const SpacePreferencesAppearanceTab: React.FC<Pick<IProps, "space">> = ({ space
                             );
                         }}
                     >
-                        {_t("People")}
+                        {_t("Direct Messages") /* TCHAP: change label _t("People") */}
                     </StyledCheckbox>
                     <SettingsSubsectionText>
                         {_t(
diff --git a/node_modules/matrix-react-sdk/src/components/views/rooms/RoomList.tsx b/node_modules/matrix-react-sdk/src/components/views/rooms/RoomList.tsx
index 64d2c73..4453a0a 100644
--- a/node_modules/matrix-react-sdk/src/components/views/rooms/RoomList.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/rooms/RoomList.tsx
@@ -402,7 +402,7 @@ const TAG_AESTHETICS: TagAestheticsMap = {
         defaultHidden: false,
     },
     [DefaultTagID.DM]: {
-        sectionLabel: _td("People"),
+        sectionLabel: _td("Direct Messages"), /* TCHAP: change label _td("People"), */
         isInvite: false,
         defaultHidden: false,
         AuxButtonComponent: DmAuxButton,
diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SidebarUserSettingsTab.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SidebarUserSettingsTab.tsx
index 3c9aaff..da2c9ad 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SidebarUserSettingsTab.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SidebarUserSettingsTab.tsx
@@ -121,7 +121,7 @@ const SidebarUserSettingsTab: React.FC = () => {
                     >
                         <SettingsSubsectionText>
                             <MembersIcon />
-                            {_t("People")}
+                            {_t("Direct Messages") /* TCHAP: change label _t("People") */}
                         </SettingsSubsectionText>
                         <SettingsSubsectionText>{_t("Group all your people in one place.")}</SettingsSubsectionText>
                     </StyledCheckbox>
diff --git a/node_modules/matrix-react-sdk/src/components/views/spaces/QuickSettingsButton.tsx b/node_modules/matrix-react-sdk/src/components/views/spaces/QuickSettingsButton.tsx
index 3966b04..68238e5 100644
--- a/node_modules/matrix-react-sdk/src/components/views/spaces/QuickSettingsButton.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/spaces/QuickSettingsButton.tsx
@@ -107,7 +107,9 @@ const QuickSettingsButton: React.FC<{
                     onChange={onMetaSpaceChangeFactory(MetaSpace.People, "WebQuickSettingsPinToSidebarCheckbox")}
                 >
                     <MembersIcon className="mx_QuickSettingsButton_icon" />
-                    {_t("People")}
+                    {/* TCHAP: change label {_t("People")} */}
+                    {_t("Direct Messages")}
+                    {/* end TCHAP */}
                 </StyledCheckbox>
                 <AccessibleButton
                     className="mx_QuickSettingsButton_moreOptionsButton"
diff --git a/node_modules/matrix-react-sdk/src/stores/spaces/index.ts b/node_modules/matrix-react-sdk/src/stores/spaces/index.ts
index fc4a3c0..bbfbfc6 100644
--- a/node_modules/matrix-react-sdk/src/stores/spaces/index.ts
+++ b/node_modules/matrix-react-sdk/src/stores/spaces/index.ts
@@ -41,7 +41,10 @@ export const getMetaSpaceName = (spaceKey: MetaSpace, allRoomsInHome = false): s
         case MetaSpace.Favourites:
             return _t("Favourites");
         case MetaSpace.People:
-            return _t("People");
+            // TCHAP:
+            // return _t("People")
+            return _t("Direct Messages");
+            // end TCHAP
         case MetaSpace.Orphans:
             return _t("Other rooms");
     }
