diff --git a/node_modules/matrix-react-sdk/src/components/views/context_menus/RoomContextMenu.tsx b/node_modules/matrix-react-sdk/src/components/views/context_menus/RoomContextMenu.tsx
index c4a3c2a..5d6d1df 100644
--- a/node_modules/matrix-react-sdk/src/components/views/context_menus/RoomContextMenu.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/context_menus/RoomContextMenu.tsx
@@ -222,7 +222,8 @@ const RoomContextMenu = ({ room, onFinished, ...props }: IProps) => {
                     onFinished();
                     PosthogTrackers.trackInteraction("WebRoomHeaderContextMenuPeopleItem", ev);
                 }}
-                label={_t("People")}
+                // label={TCHAP: change label _t("People")}
+                label={_t("Direct Messages") /* end TCHAP */}
                 iconClassName="mx_RoomTile_iconPeople"
             >
                 <span className="mx_IconizedContextMenu_sublabel">{room.getJoinedMemberCount()}</span>
diff --git a/node_modules/matrix-react-sdk/src/components/views/dialogs/SpacePreferencesDialog.tsx b/node_modules/matrix-react-sdk/src/components/views/dialogs/SpacePreferencesDialog.tsx
index 61c9de1..5833e76 100644
--- a/node_modules/matrix-react-sdk/src/components/views/dialogs/SpacePreferencesDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/dialogs/SpacePreferencesDialog.tsx
@@ -52,7 +52,9 @@ const SpacePreferencesAppearanceTab = ({ space }: Pick<IProps, "space">) => {
                         );
                     }}
                 >
-                    {_t("People")}
+                    {/* TCHAP: change label {_t("People")} */}
+                    {_t("Direct Messages")}
+                    {/* end TCHAP */}
                 </StyledCheckbox>
                 <p>
                     {_t(
diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SidebarUserSettingsTab.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SidebarUserSettingsTab.tsx
index 4518fcf..845641a 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SidebarUserSettingsTab.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SidebarUserSettingsTab.tsx
@@ -105,7 +105,9 @@ const SidebarUserSettingsTab = () => {
                     onChange={onMetaSpaceChangeFactory(MetaSpace.People, "WebSettingsSidebarTabSpacesCheckbox")}
                     className="mx_SidebarUserSettingsTab_peopleCheckbox"
                 >
-                    {_t("People")}
+                    {/* TCHAP: change label {_t("People")} */}
+                    {_t("Direct Messages")}
+                    {/* end TCHAP */}
                 </StyledCheckbox>
                 <div className="mx_SidebarUserSettingsTab_checkboxMicrocopy">
                     {_t("Group all your people in one place.")}
diff --git a/node_modules/matrix-react-sdk/src/components/views/spaces/QuickSettingsButton.tsx b/node_modules/matrix-react-sdk/src/components/views/spaces/QuickSettingsButton.tsx
index 2402907..6b81b41 100644
--- a/node_modules/matrix-react-sdk/src/components/views/spaces/QuickSettingsButton.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/spaces/QuickSettingsButton.tsx
@@ -103,7 +103,9 @@ const QuickSettingsButton = ({ isPanelCollapsed = false }) => {
                     onChange={onMetaSpaceChangeFactory(MetaSpace.People, "WebQuickSettingsPinToSidebarCheckbox")}
                 >
                     <MembersIcon className="mx_QuickSettingsButton_icon" />
-                    {_t("People")}
+                    {/* TCHAP: change label {_t("People")} */}
+                    {_t("Direct Messages")}
+                    {/* end TCHAP */}
                 </StyledCheckbox>
                 <AccessibleButton
                     className="mx_QuickSettingsButton_moreOptionsButton"
diff --git a/node_modules/matrix-react-sdk/src/stores/spaces/index.ts b/node_modules/matrix-react-sdk/src/stores/spaces/index.ts
index 77d65c1..d08c798 100644
--- a/node_modules/matrix-react-sdk/src/stores/spaces/index.ts
+++ b/node_modules/matrix-react-sdk/src/stores/spaces/index.ts
@@ -42,7 +42,10 @@ export const getMetaSpaceName = (spaceKey: MetaSpace, allRoomsInHome = false): s
         case MetaSpace.Favourites:
             return _t("Favourites");
         case MetaSpace.People:
-            return _t("People");
+            // TCHAP:
+            // return _t("People")
+            return _t("Direct Messages");
+            // end TCHAP
         case MetaSpace.Orphans:
             return _t("Other rooms");
     }
