diff --git a/node_modules/matrix-react-sdk/res/css/views/auth/_CompleteSecurityBody.pcss b/node_modules/matrix-react-sdk/res/css/views/auth/_CompleteSecurityBody.pcss
index c23f53b..7aa9f25 100644
--- a/node_modules/matrix-react-sdk/res/css/views/auth/_CompleteSecurityBody.pcss
+++ b/node_modules/matrix-react-sdk/res/css/views/auth/_CompleteSecurityBody.pcss
@@ -16,7 +16,12 @@ limitations under the License.
 */
 
 .mx_CompleteSecurityBody {
+    /* :TCHAP:
     width: 600px;
+    */
+    width: 660px;
+    /* end :TCHAP: */
+
     color: $authpage-primary-color;
     background-color: $background;
     border-radius: 4px;
diff --git a/node_modules/matrix-react-sdk/src/async-components/views/dialogs/security/CreateSecretStorageDialog.tsx b/node_modules/matrix-react-sdk/src/async-components/views/dialogs/security/CreateSecretStorageDialog.tsx
index b595a60..929ad9a 100644
--- a/node_modules/matrix-react-sdk/src/async-components/views/dialogs/security/CreateSecretStorageDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/async-components/views/dialogs/security/CreateSecretStorageDialog.tsx
@@ -177,13 +177,38 @@ export default class CreateSecretStorageDialog extends React.PureComponent<IProp
                 MatrixClientPeg.get().isCryptoEnabled() && (await MatrixClientPeg.get().isKeyBackupTrusted(backupInfo));
 
             const { forceReset } = this.props;
-            const phase = backupInfo && !forceReset ? Phase.Migrate : Phase.ChooseKeyPassphrase;
+            // :tchap: const phase = backupInfo && !forceReset ? Phase.Migrate : Phase.ChooseKeyPassphrase;
+            const phase = backupInfo && !forceReset ? Phase.Migrate : Phase.ShowKey;//:tchap: goes directly to showkey
 
+            /* :TCHAP: remove
             this.setState({
                 phase,
                 backupInfo,
                 backupSigStatus,
             });
+            end :TCHAP: */
+
+            // add :TCHAP:
+            if (phase === Phase.ShowKey) {
+                this.recoveryKey = await MatrixClientPeg.get().createRecoveryKeyFromPassphrase();
+                this.setState({
+                    phase,
+                    backupInfo,
+                    backupSigStatus,
+                    passPhraseKeySelected:SecureBackupSetupMethod.Key,
+                    copied: false,
+                    downloaded: false,
+                    setPassphrase: false 
+                });
+            } else {
+                //if phase is Phase.Migrate
+                this.setState({
+                    phase,
+                    backupInfo,
+                    backupSigStatus,
+                });
+            }
+            // end :TCHAP:
 
             return {
                 backupInfo,
@@ -719,7 +744,10 @@ export default class CreateSecretStorageDialog extends React.PureComponent<IProp
         if (this.state.phase === Phase.ShowKey) {
             continueButton = (
                 <DialogButtons
-                    primaryButton={_t("Continue")}
+                    // :tchap: change label
+                    // primaryButton={_t("Continue")}
+                    primaryButton={_t("Activate")}
+                    // end :tchap:
                     disabled={!this.state.downloaded && !this.state.copied && !this.state.setPassphrase}
                     onPrimaryButtonClick={this.onShowKeyContinueClick}
                     hasCancel={false}
@@ -735,18 +763,27 @@ export default class CreateSecretStorageDialog extends React.PureComponent<IProp
 
         return (
             <div>
+                {/* tchap: add this <p /> */}
+                <p>
+                    {_t(
+                        "This is your recovery key"
+                    )}
+                </p>
+                {/* end tchap */}
                 <p>
                     {_t(
                         "Store your Security Key somewhere safe, like a password manager or a safe, " +
                             "as it's used to safeguard your encrypted data.",
                     )}
                 </p>
+
                 <div className="mx_CreateSecretStorageDialog_primaryContainer mx_CreateSecretStorageDialog_recoveryKeyPrimarycontainer">
                     <div className="mx_CreateSecretStorageDialog_recoveryKeyContainer">
                         <div className="mx_CreateSecretStorageDialog_recoveryKey">
                             <code ref={this.recoveryKeyNode}>{this.recoveryKey.encodedPrivateKey}</code>
                         </div>
                         <div className="mx_CreateSecretStorageDialog_recoveryKeyButtons">
+                            {/* :TCHAP: remove
                             <AccessibleButton
                                 kind="primary"
                                 className="mx_Dialog_primary"
@@ -761,6 +798,7 @@ export default class CreateSecretStorageDialog extends React.PureComponent<IProp
                                     copyButton: "",
                                 })}
                             </span>
+                            end :TCHAP: */}
                             <AccessibleButton
                                 kind="primary"
                                 className="mx_Dialog_primary mx_CreateSecretStorageDialog_recoveryKeyButtons_copyBtn"
diff --git a/node_modules/matrix-react-sdk/src/components/structures/auth/CompleteSecurity.tsx b/node_modules/matrix-react-sdk/src/components/structures/auth/CompleteSecurity.tsx
index 7b898d6..b386b44 100644
--- a/node_modules/matrix-react-sdk/src/components/structures/auth/CompleteSecurity.tsx
+++ b/node_modules/matrix-react-sdk/src/components/structures/auth/CompleteSecurity.tsx
@@ -22,6 +22,7 @@ import SetupEncryptionBody from "./SetupEncryptionBody";
 import AccessibleButton from "../../views/elements/AccessibleButton";
 import CompleteSecurityBody from "../../views/auth/CompleteSecurityBody";
 import AuthPage from "../../views/auth/AuthPage";
+import TchapUIFeature from "../../../../../../src/util/TchapUIFeature";
 
 interface IProps {
     onFinished: () => void;
@@ -92,9 +93,16 @@ export default class CompleteSecurity extends React.Component<IProps, IState> {
 
         let skipButton;
         if (phase === Phase.Intro || phase === Phase.ConfirmReset) {
+            // :Tchap: Condition to skip Phase.ConfirmSkip and its "Are you sure" modal after login for csss
+            const tchapOnSkipClick = phase === Phase.Intro ? this.props.onFinished : this.onSkipClick;
+            // end :Tchap:
+
             skipButton = (
                 <AccessibleButton
-                    onClick={this.onSkipClick}
+                    // :tchap: remove onClick={this.onSkipClick}
+                    // add instead
+                    onClick={TchapUIFeature.isCrossSigningAndSecureStorageActive ? tchapOnSkipClick : this.onSkipClick}
+                    // end :tchap:
                     className="mx_CompleteSecurity_skip"
                     aria-label={_t("Skip verification for now")}
                 />
