diff --git a/node_modules/matrix-react-sdk/res/css/views/elements/_TooltipButton.pcss b/node_modules/matrix-react-sdk/res/css/views/elements/_TooltipButton.pcss
index cc91c6d..0cbbf73 100644
--- a/node_modules/matrix-react-sdk/res/css/views/elements/_TooltipButton.pcss
+++ b/node_modules/matrix-react-sdk/res/css/views/elements/_TooltipButton.pcss
@@ -15,6 +15,8 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+@import "../../../../../../res/css/views/elements/_TchapTooltipButton.pcss"; // :TCHAP:
+
 .mx_TooltipButton {
     display: inline-block;
     width: 11px;
diff --git a/node_modules/matrix-react-sdk/src/components/views/rooms/EventTile.tsx b/node_modules/matrix-react-sdk/src/components/views/rooms/EventTile.tsx
index 28a6fff..188c1c9 100644
--- a/node_modules/matrix-react-sdk/src/components/views/rooms/EventTile.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/rooms/EventTile.tsx
@@ -86,6 +86,7 @@ import { ShowThreadPayload } from "../../../dispatcher/payloads/ShowThreadPayloa
 import { isLocalRoom } from "../../../utils/localRoom/isLocalRoom";
 import { ElementCall } from "../../../models/Call";
 import { UnreadNotificationBadge } from "./NotificationBadge/UnreadNotificationBadge";
+import { TchapKeyRequestInfo } from "../../../../../../src/components/views/rooms/TchapKeyRequestInfo"; // :TCHAP:
 
 export type GetRelationsForEvent = (
     eventId: string,
@@ -1543,7 +1544,12 @@ export class UnwrappedEventTile extends React.Component<EventTileProps, IState>
                                 },
                                 this.context.showHiddenEvents,
                             )}
-                            {keyRequestInfo}
+                            <TchapKeyRequestInfo // :TCHAP:
+                                previouslyRequestedKeys={this.state.previouslyRequestedKeys}
+                                onRequestKeysClick={this.onRequestKeysClick}
+                                isEncryptionFailure={isEncryptionFailure}
+                                isRedacted={isRedacted}>
+                            </TchapKeyRequestInfo>
                             {actionBar}
                             {this.props.layout === Layout.IRC && (
                                 <>
