diff --git a/node_modules/matrix-react-sdk/src/components/views/messages/UnknownBody.tsx b/node_modules/matrix-react-sdk/src/components/views/messages/UnknownBody.tsx
index cd1f06a..434c949 100644
--- a/node_modules/matrix-react-sdk/src/components/views/messages/UnknownBody.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/messages/UnknownBody.tsx
@@ -17,13 +17,26 @@ limitations under the License.
 
 import React, { forwardRef } from "react";
 import { MatrixEvent } from "matrix-js-sdk/src/matrix";
-
+import { _t } from 'matrix-react-sdk/src/languageHandler'; // :TCHAP:
 interface IProps {
     mxEvent: MatrixEvent;
     children?: React.ReactNode;
 }
 
 export default forwardRef(({ mxEvent, children }: IProps, ref: React.RefObject<HTMLDivElement>) => {
+    // :TCHAP: user-friendly message in the case of lost keys
+    const content = mxEvent.getContent();
+    if (content.msgtype && content.msgtype === "m.bad.encrypted") {
+        const userFriendlyText = _t("Decryption fail: Please open Tchap on an other connected device to allow key sharing.");
+        return (
+            <div className="mx_UnknownBody" ref={ref}>
+                { userFriendlyText }
+                { children }
+            </div>
+        );
+    }
+    // end :TCHAP:
+
     const text = mxEvent.getContent().body;
     return (
         <div className="mx_UnknownBody" ref={ref}>
diff --git a/node_modules/matrix-react-sdk/src/components/views/rooms/EventTile.tsx b/node_modules/matrix-react-sdk/src/components/views/rooms/EventTile.tsx
index 654cc80..6d3d222 100644
--- a/node_modules/matrix-react-sdk/src/components/views/rooms/EventTile.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/rooms/EventTile.tsx
@@ -1167,7 +1167,10 @@ export class UnwrappedEventTile extends React.Component<IProps, IState> {
         const keyRequestInfoContent = this.state.previouslyRequestedKeys ?
             _t('Key request sent.') :
             _t(
-                '<requestLink>Re-request encryption keys</requestLink> from your other sessions.',
+                // :TCHAP: better text
+                //'<requestLink>Re-request encryption keys</requestLink> from your other sessions.',
+                '<requestLink>Re-send a request to your other devices</requestLink>',
+                // end :TCHAP:
                 {},
                 {
                     'requestLink': (sub) =>
