diff --git a/node_modules/matrix-react-sdk/src/components/views/dialogs/IncomingSasDialog.tsx b/node_modules/matrix-react-sdk/src/components/views/dialogs/IncomingSasDialog.tsx
index b0bb9cb..3b440e4 100644
--- a/node_modules/matrix-react-sdk/src/components/views/dialogs/IncomingSasDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/dialogs/IncomingSasDialog.tsx
@@ -179,6 +179,7 @@ export default class IncomingSasDialog extends React.Component<IProps, IState> {
             profile = <Spinner />;
         }
 
+        /* :TCHAP: remove code
         const userDetailText = [
             <p key="p1">
                 {_t(
@@ -212,11 +213,23 @@ export default class IncomingSasDialog extends React.Component<IProps, IState> {
                 )}
             </p>,
         ];
+        end :TCHAP: */
 
         return (
             <div>
+                {/* :TCHAP: remove code
                 {profile}
                 {isSelf ? selfDetailText : userDetailText}
+                end :TCHAP: */}
+
+                {/* :TCHAP: simplify modal message */}
+                <p>{ _t(
+                    "One of your devices <b>wants to check your Tchap Keys</b> to unlock your messages.",
+                    {},
+                    { b: sub => <b>{ sub }</b> },
+                ) }</p>
+                {/* end :TCHAP: */}
+
                 <DialogButtons
                     primaryButton={_t("Continue")}
                     hasCancel={true}
@@ -244,7 +257,10 @@ export default class IncomingSasDialog extends React.Component<IProps, IState> {
         return (
             <div>
                 <Spinner />
+                {/* :TCHAP: remove code
                 <p>{_t("Waiting for partner to confirm…")}</p>
+                end :TCHAP: */}
+                <p>{ _t("Please confirm the objects on the other device (loading).")}</p>
             </div>
         );
     }
diff --git a/node_modules/matrix-react-sdk/src/components/views/dialogs/VerificationRequestDialog.tsx b/node_modules/matrix-react-sdk/src/components/views/dialogs/VerificationRequestDialog.tsx
index 53ac14f..dd600f8 100644
--- a/node_modules/matrix-react-sdk/src/components/views/dialogs/VerificationRequestDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/dialogs/VerificationRequestDialog.tsx
@@ -49,7 +49,10 @@ export default class VerificationRequestDialog extends React.Component<IProps, I
         const request = this.state.verificationRequest;
         const otherUserId = request?.otherUserId;
         const member = this.props.member || (otherUserId ? MatrixClientPeg.get().getUser(otherUserId) : null);
+        /* :TCHAP: remove code
         const title = request?.isSelfVerification ? _t("Verify other device") : _t("Verification Request");
+        end :TCHAP: */
+        const title = _t("Incoming Verification Request");
         
         return (
             <BaseDialog
diff --git a/node_modules/matrix-react-sdk/src/components/views/right_panel/VerificationPanel.tsx b/node_modules/matrix-react-sdk/src/components/views/right_panel/VerificationPanel.tsx
index 1820309..96f34db 100644
--- a/node_modules/matrix-react-sdk/src/components/views/right_panel/VerificationPanel.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/right_panel/VerificationPanel.tsx
@@ -278,10 +278,13 @@ export default class VerificationPanel extends React.PureComponent<IProps, IStat
                 logger.warn("Verified device we don't know about: " + this.props.request.channel.deviceId);
                 description = _t("You've successfully verified your device!");
             } else {
+                description = _t("The sharing of your Tchap Keys has succeeded. Your messages will be unlocked.");
+                /* :TCHAP:
                 description = _t("You've successfully verified %(deviceName)s (%(deviceId)s)!", {
                     deviceName: device ? device.getDisplayName() : "",
                     deviceId: this.props.request.channel.deviceId,
                 });
+                end :TCHAP: */
             }
         } else {
             description = _t("You've successfully verified %(displayName)s!", {
diff --git a/node_modules/matrix-react-sdk/src/components/views/verification/VerificationComplete.tsx b/node_modules/matrix-react-sdk/src/components/views/verification/VerificationComplete.tsx
index 1f034b4..6d43689 100644
--- a/node_modules/matrix-react-sdk/src/components/views/verification/VerificationComplete.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/verification/VerificationComplete.tsx
@@ -27,6 +27,9 @@ export default class VerificationComplete extends React.Component<IProps> {
     public render(): React.ReactNode {
         return (
             <div>
+                <h2>{ _t("Incoming Verification Request") }</h2>
+                <p>{ _t("The sharing of your Tchap Keys has succeeded. Your messages will be unlocked.") }</p>
+                {/* :TCHAP: remove code
                 <h2>{_t("Verified!")}</h2>
                 <p>{_t("You've successfully verified this user.")}</p>
                 <p>
@@ -35,6 +38,7 @@ export default class VerificationComplete extends React.Component<IProps> {
                             "read by third parties.",
                     )}
                 </p>
+                end :TCHAP: */}
                 <DialogButtons
                     onPrimaryButtonClick={this.props.onDone}
                     primaryButton={_t("Got It")}
