diff --git a/node_modules/matrix-react-sdk/src/components/views/dialogs/BugReportDialog.tsx b/node_modules/matrix-react-sdk/src/components/views/dialogs/BugReportDialog.tsx
index 7b389a3..a2b0a54 100644
--- a/node_modules/matrix-react-sdk/src/components/views/dialogs/BugReportDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/dialogs/BugReportDialog.tsx
@@ -32,6 +32,7 @@ import DialogButtons from "../elements/DialogButtons";
 import { sendSentryReport } from "../../../sentry";
 import defaultDispatcher from "../../../dispatcher/dispatcher";
 import { Action } from "../../../dispatcher/actions";
+import { MatrixClientPeg } from '../../../MatrixClientPeg';  // :TCHAP:
 
 interface IProps {
     onFinished: (success: boolean) => void;
@@ -88,12 +89,21 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
     };
 
     private onSubmit = (): void => {
+        /* :TCHAP: do not ask for a github issue
         if ((!this.state.text || !this.state.text.trim()) && (!this.state.issueUrl || !this.state.issueUrl.trim())) {
             this.setState({
                 err: _t("Please tell us what went wrong or, better, create a GitHub issue that describes the problem."),
             });
             return;
         }
+        */
+        if ((!this.state.text || !this.state.text.trim())) {
+            this.setState({
+                err: _t("Please tell us what went wrong in the \"Notes\" field."),
+            });
+            return;
+        }
+        // end :TCHAP:
 
         const userText =
             (this.state.text.length > 0 ? this.state.text + "\n\n" : "") +
@@ -103,11 +113,24 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
         this.setState({ busy: true, progress: null, err: null });
         this.sendProgressCallback(_t("Preparing to send logs"));
 
-        sendBugReport(SdkConfig.get().bug_report_endpoint_url, {
-            userText,
-            sendLogs: true,
-            progressCallback: this.sendProgressCallback,
-            labels: this.props.label ? [this.props.label] : [],
+        // :TCHAP: customise report : add email, prefix with "tchap-web"
+        const client = MatrixClientPeg.get();
+        client.getThreePids().then(result => {
+            const customFields = {};
+            result.threepids.forEach(threepid => {
+                return customFields[threepid.medium] = threepid.address;
+            });
+            return customFields;
+        }).then(customFields => {
+            return sendBugReport(SdkConfig.get().bug_report_endpoint_url, {
+                userText,
+                sendLogs: true,
+                progressCallback: this.sendProgressCallback,
+                labels: this.props.label ? [this.props.label] : [],
+                customApp: 'tchap-web', // :TCHAP:
+                customFields: customFields, // :TCHAP:
+            });
+            // end :TCHAP:
         }).then(
             () => {
                 if (!this.unmounted) {
@@ -142,6 +165,7 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
                 sendLogs: true,
                 progressCallback: this.downloadProgressCallback,
                 labels: this.props.label ? [this.props.label] : [],
+                customApp: 'tchap-web', // :TCHAP: we don't add email here. You know your own email already.
             });
 
             this.setState({
@@ -209,7 +233,7 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
             <BaseDialog
                 className="mx_BugReportDialog"
                 onFinished={this.onCancel}
-                title={_t("Submit debug logs")}
+                title={_t('Submit debug logs to Tchap support team') /** :TCHAP: more precise title than "Submit debug logs"*/}
                 contentId="mx_Dialog_content"
             >
                 <div className="mx_Dialog_content" id="mx_Dialog_content">
@@ -222,6 +246,7 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
                                 "and the usernames of other users. They do not contain messages.",
                         )}
                     </p>
+                    { /* :TCHAP: remove requirement to send github issue, it's too complicated
                     <p>
                         <b>
                             {_t(
@@ -231,7 +256,9 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
                                     a: (sub) => (
                                         <a
                                             target="_blank"
-                                            href="https://github.com/vector-im/element-web/issues/new/choose"
+                                            // :TCHAP: href="https://github.com/vector-im/element-web/issues/new/choose"
+                                            href="https://github.com/tchapgouv/tchap-web-v4/issues/new/choose"
+                                            // :end TCHAP:
                                         >
                                             {sub}
                                         </a>
@@ -240,22 +267,28 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
                             )}
                         </b>
                     </p>
+                    */}
 
+                    { /* :TCHAP: move this below, it's not where the user expects it, so it's hard to understand
                     <div className="mx_BugReportDialog_download">
                         <AccessibleButton onClick={this.onDownload} kind="link" disabled={this.state.downloadBusy}>
                             {_t("Download logs")}
                         </AccessibleButton>
                         {this.state.downloadProgress && <span>{this.state.downloadProgress} ...</span>}
                     </div>
+                    */ }
 
+                    { /* :TCHAP: remove requirement to send github issue, it's too complicated
                     <Field
                         type="text"
                         className="mx_BugReportDialog_field_input"
                         label={_t("GitHub issue")}
                         onChange={this.onIssueUrlChange}
                         value={this.state.issueUrl}
-                        placeholder="https://github.com/vector-im/element-web/issues/..."
-                    />
+                        // :TCHAP: placeholder="https://github.com/vector-im/element-web/issues/..."
+                        placeholder="https://github.com/tchapgouv/tchap-web-v4/issues/..."
+                        // end :TCHAP:                    />
+                    */ }
                     <Field
                         className="mx_BugReportDialog_field_input"
                         element="textarea"
@@ -272,14 +305,29 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
                     />
                     {progress}
                     {error}
+                    <DialogButtons primaryButton={_t("Send logs")}
+                        onPrimaryButtonClick={this.onSubmit}
+                        focus={true}
+                        // :TCHAP: remove cancel button, now that we added more buttons it doesn't make sense
+                        // onCancel={this.onCancel}
+                        hasCancel={false}
+                        disabled={this.state.busy}
+                    />
+
+                    {/** :TCHAP: move the log download here, and add an explanation text */}
+                    <div className="mx_BugReportDialog_download mx_Dialog_content">
+                        <p>
+                            { _t("Just want to get your own logs, without sharing them with the Tchap team?") }
+                        </p>
+                        <div className="mx_Dialog_buttons_row">
+                            { this.state.downloadProgress && <span>{ this.state.downloadProgress } ...</span> }
+                            <AccessibleButton onClick={this.onDownload} kind="primary" disabled={this.state.downloadBusy}>
+                                { _t("Download my logs") }
+                            </AccessibleButton>
+                        </div>
+                    </div>
                 </div>
-                <DialogButtons
-                    primaryButton={_t("Send logs")}
-                    onPrimaryButtonClick={this.onSubmit}
-                    focus={true}
-                    onCancel={this.onCancel}
-                    disabled={this.state.busy}
-                />
+                {/** end :TCHAP: */}
             </BaseDialog>
         );
     }
