diff --git a/node_modules/matrix-react-sdk/src/components/views/dialogs/BugReportDialog.tsx b/node_modules/matrix-react-sdk/src/components/views/dialogs/BugReportDialog.tsx
index ed706f9..8b14d80 100644
--- a/node_modules/matrix-react-sdk/src/components/views/dialogs/BugReportDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/dialogs/BugReportDialog.tsx
@@ -32,6 +32,7 @@ import DialogButtons from "../elements/DialogButtons";
 import { sendSentryReport } from "../../../sentry";
 import defaultDispatcher from "../../../dispatcher/dispatcher";
 import { Action } from "../../../dispatcher/actions";
+import { MatrixClientPeg } from '../../../MatrixClientPeg';  // :TCHAP:
 
 interface IProps {
     onFinished: (success: boolean) => void;
@@ -96,12 +97,21 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
     };
 
     private onSubmit = (): void => {
+        /* :TCHAP: do not ask for a github issue
         if ((!this.state.text || !this.state.text.trim()) && (!this.state.issueUrl || !this.state.issueUrl.trim())) {
             this.setState({
                 err: _t("Please tell us what went wrong or, better, create a GitHub issue that describes the problem."),
             });
             return;
         }
+        */
+        if ((!this.state.text || !this.state.text.trim())) {
+            this.setState({
+                err: _t("Please tell us what went wrong in the \"Notes\" field."),
+            });
+            return;
+        }
+        // end :TCHAP:
 
         const userText =
             (this.state.text.length > 0 ? this.state.text + "\n\n" : "") +
@@ -111,11 +121,24 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
         this.setState({ busy: true, progress: null, err: null });
         this.sendProgressCallback(_t("Preparing to send logs"));
 
-        sendBugReport(SdkConfig.get().bug_report_endpoint_url, {
-            userText,
-            sendLogs: true,
-            progressCallback: this.sendProgressCallback,
-            labels: this.props.label ? [this.props.label] : [],
+        // :TCHAP: customise report : add email, prefix with "tchap-web"
+        const client = MatrixClientPeg.get();
+        client.getThreePids().then(result => {
+            const customFields = {};
+            result.threepids.forEach(threepid => {
+                return customFields[threepid.medium] = threepid.address;
+            });
+            return customFields;
+        }).then(customFields => {
+            return sendBugReport(SdkConfig.get().bug_report_endpoint_url, {
+                userText,
+                sendLogs: true,
+                progressCallback: this.sendProgressCallback,
+                labels: this.props.label ? [this.props.label] : [],
+                customApp: 'tchap-web', // :TCHAP:
+                customFields: customFields, // :TCHAP:
+            });
+            // end :TCHAP:
         }).then(
             () => {
                 if (!this.unmounted) {
@@ -150,6 +173,7 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
                 sendLogs: true,
                 progressCallback: this.downloadProgressCallback,
                 labels: this.props.label ? [this.props.label] : [],
+                customApp: 'tchap-web', // :TCHAP: we don't add email here. You know your own email already.
             });
 
             this.setState({
@@ -217,7 +241,7 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
             <BaseDialog
                 className="mx_BugReportDialog"
                 onFinished={this.onCancel}
-                title={_t("Submit debug logs")}
+                title={_t('Submit debug logs to Tchap support team') /** :TCHAP: more precise title than "Submit debug logs"*/}
                 contentId="mx_Dialog_content"
             >
                 <div className="mx_Dialog_content" id="mx_Dialog_content">
@@ -230,6 +254,7 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
                                 "and the usernames of other users. They do not contain messages.",
                         )}
                     </p>
+                    { /* :TCHAP: remove requirement to send github issue, it's too complicated
                     <p>
                         <b>
                             {_t(
@@ -239,7 +264,9 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
                                     a: (sub) => (
                                         <a
                                             target="_blank"
-                                            href="https://github.com/vector-im/element-web/issues/new/choose"
+                                            // :TCHAP: href="https://github.com/vector-im/element-web/issues/new/choose"
+                                            href="https://github.com/tchapgouv/tchap-web-v4/issues/new/choose"
+                                            // :end TCHAP:
                                         >
                                             {sub}
                                         </a>
@@ -248,23 +275,28 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
                             )}
                         </b>
                     </p>
+                    */}
 
+                    { /* :TCHAP: move this below, it's not where the user expects it, so it's hard to understand
                     <div className="mx_BugReportDialog_download">
                         <AccessibleButton onClick={this.onDownload} kind="link" disabled={this.state.downloadBusy}>
                             {_t("Download logs")}
                         </AccessibleButton>
                         {this.state.downloadProgress && <span>{this.state.downloadProgress} ...</span>}
                     </div>
+                    */ }
 
+                    { /* :TCHAP: remove requirement to send github issue, it's too complicated
                     <Field
                         type="text"
                         className="mx_BugReportDialog_field_input"
                         label={_t("GitHub issue")}
                         onChange={this.onIssueUrlChange}
                         value={this.state.issueUrl}
-                        placeholder="https://github.com/vector-im/element-web/issues/..."
-                        ref={this.issueRef}
-                    />
+                        // :TCHAP: placeholder="https://github.com/vector-im/element-web/issues/..."
+                        placeholder="https://github.com/tchapgouv/tchap-web-v4/issues/..."
+                        // end :TCHAP:                    />
+                    */ }
                     <Field
                         className="mx_BugReportDialog_field_input"
                         element="textarea"
@@ -281,14 +313,29 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
                     />
                     {progress}
                     {error}
+                    <DialogButtons primaryButton={_t("Send logs")}
+                        onPrimaryButtonClick={this.onSubmit}
+                        focus={true}
+                        // :TCHAP: remove cancel button, now that we added more buttons it doesn't make sense
+                        // onCancel={this.onCancel}
+                        hasCancel={false}
+                        disabled={this.state.busy}
+                    />
+
+                    {/** :TCHAP: move the log download here, and add an explanation text */}
+                    <div className="mx_BugReportDialog_download mx_Dialog_content">
+                        <p>
+                            { _t("Just want to get your own logs, without sharing them with the Tchap team?") }
+                        </p>
+                        <div className="mx_Dialog_buttons_row">
+                            { this.state.downloadProgress && <span>{ this.state.downloadProgress } ...</span> }
+                            <AccessibleButton onClick={this.onDownload} kind="primary" disabled={this.state.downloadBusy}>
+                                { _t("Download my logs") }
+                            </AccessibleButton>
+                        </div>
+                    </div>
                 </div>
-                <DialogButtons
-                    primaryButton={_t("Send logs")}
-                    onPrimaryButtonClick={this.onSubmit}
-                    focus={true}
-                    onCancel={this.onCancel}
-                    disabled={this.state.busy}
-                />
+                {/** end :TCHAP: */}
             </BaseDialog>
         );
     }
diff --git a/node_modules/matrix-react-sdk/src/components/views/dialogs/FeedbackDialog.tsx b/node_modules/matrix-react-sdk/src/components/views/dialogs/FeedbackDialog.tsx
index 7ee24e0..6a9d898 100644
--- a/node_modules/matrix-react-sdk/src/components/views/dialogs/FeedbackDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/dialogs/FeedbackDialog.tsx
@@ -28,9 +28,13 @@ import { submitFeedback } from "../../../rageshake/submit-rageshake";
 import { useStateToggle } from "../../../hooks/useStateToggle";
 import StyledCheckbox from "../elements/StyledCheckbox";
 
-const existingIssuesUrl =
-    "https://github.com/vector-im/element-web/issues" + "?q=is%3Aopen+is%3Aissue+sort%3Areactions-%2B1-desc";
-const newIssueUrl = "https://github.com/vector-im/element-web/issues/new/choose";
+// :TCHAP: change github urls
+// const existingIssuesUrl =
+//     "https://github.com/vector-im/element-web/issues" + "?q=is%3Aopen+is%3Aissue+sort%3Areactions-%2B1-desc";
+// const newIssueUrl = "https://github.com/vector-im/element-web/issues/new/choose";
+const existingIssuesUrl = "https://github.com/tchapgouv/tchap-web-v4/issues" + "?q=is%3Aopen+is%3Aissue+sort%3Areactions-%2B1-desc";
+const newIssueUrl = "https://github.com/tchapgouv/tchap-web-v4/issues/new/choose";
+// end :TCHAP:
 
 interface IProps {
     feature?: string;
diff --git a/node_modules/matrix-react-sdk/src/components/views/elements/AccessibleButton.tsx b/node_modules/matrix-react-sdk/src/components/views/elements/AccessibleButton.tsx
index 10f18f1..8d18769 100644
--- a/node_modules/matrix-react-sdk/src/components/views/elements/AccessibleButton.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/elements/AccessibleButton.tsx
@@ -75,7 +75,7 @@ type IProps<T extends keyof JSX.IntrinsicElements> = DynamicHtmlElementProps<T>
     disabled?: boolean;
     className?: string;
     triggerOnMouseDown?: boolean;
-    onClick: ((e: ButtonEvent) => void | Promise<void>) | null;
+    onClick?: ((e: ButtonEvent) => void | Promise<void>) | null;
 };
 
 export interface IAccessibleButtonProps extends React.InputHTMLAttributes<Element> {
diff --git a/node_modules/matrix-react-sdk/src/components/views/elements/ErrorBoundary.tsx b/node_modules/matrix-react-sdk/src/components/views/elements/ErrorBoundary.tsx
index a230ab8..5a9a2a2 100644
--- a/node_modules/matrix-react-sdk/src/components/views/elements/ErrorBoundary.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/elements/ErrorBoundary.tsx
@@ -77,7 +77,9 @@ export default class ErrorBoundary extends React.PureComponent<Props, IState> {
 
     public render(): ReactNode {
         if (this.state.error) {
-            const newIssueUrl = "https://github.com/vector-im/element-web/issues/new/choose";
+            // :TCHAP: change github url
+            // const newIssueUrl = "https://github.com/vector-im/element-web/issues/new/choose";
+            const newIssueUrl = "https://github.com/tchapgouv/tchap-web-v4/issues/new/choose";
 
             let bugReportSection;
             if (SdkConfig.get().bug_report_endpoint_url) {
diff --git a/node_modules/matrix-react-sdk/src/components/views/rooms/RoomPreviewBar.tsx b/node_modules/matrix-react-sdk/src/components/views/rooms/RoomPreviewBar.tsx
index c3f75c1..9019ff1 100644
--- a/node_modules/matrix-react-sdk/src/components/views/rooms/RoomPreviewBar.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/rooms/RoomPreviewBar.tsx
@@ -563,7 +563,9 @@ export default class RoomPreviewBar extends React.Component<IProps, IState> {
                         {
                             issueLink: (label) => (
                                 <a
-                                    href="https://github.com/vector-im/element-web/issues/new/choose"
+                                    // :TCHAP: change github url
+                                    // href="https://github.com/vector-im/element-web/issues/new/choose"
+                                    href="https://github.com/tchapgouv/tchap-web-v4/issues/new/choose"
                                     target="_blank"
                                     rel="noreferrer noopener"
                                 >
diff --git a/node_modules/matrix-react-sdk/src/rageshake/submit-rageshake.ts b/node_modules/matrix-react-sdk/src/rageshake/submit-rageshake.ts
index 9027e28..791c1c6 100644
--- a/node_modules/matrix-react-sdk/src/rageshake/submit-rageshake.ts
+++ b/node_modules/matrix-react-sdk/src/rageshake/submit-rageshake.ts
@@ -291,12 +291,13 @@ export async function submitFeedback(
         version = await PlatformPeg.get()?.getAppVersion();
     } catch (err) {} // PlatformPeg already logs this.
 
+
     const body = new FormData();
     body.append("label", label);
     body.append("text", comment);
     body.append("can_contact", canContact ? "yes" : "no");
 
-    body.append("app", "element-web");
+    body.append("app", "tchap-web");
     body.append("version", version || "UNKNOWN");
     body.append("platform", PlatformPeg.get()?.getHumanReadableName() ?? "n/a");
     body.append("user_id", MatrixClientPeg.get()?.getUserId() ?? "n/a");
@@ -306,6 +307,13 @@ export async function submitFeedback(
     }
 
     const bugReportEndpointUrl = SdkConfig.get().bug_report_endpoint_url;
+    //:tchap: add email in body
+    const client = MatrixClientPeg.get();
+    const result = await client.getThreePids();//it generates a API calls which is acceptable because feedbacks submit are not so frequent (unfortunately)
+    result.threepids.forEach(threepid => {
+        body.append(threepid.medium, threepid.address);
+    });
+    //:tchap: end
 
     if (bugReportEndpointUrl) {
         await submitReport(bugReportEndpointUrl, body, () => {});
