diff --git a/node_modules/matrix-react-sdk/res/css/views/dialogs/_InviteDialog.pcss b/node_modules/matrix-react-sdk/res/css/views/dialogs/_InviteDialog.pcss
index c03e78d..8f7854d 100644
--- a/node_modules/matrix-react-sdk/res/css/views/dialogs/_InviteDialog.pcss
+++ b/node_modules/matrix-react-sdk/res/css/views/dialogs/_InviteDialog.pcss
@@ -38,7 +38,7 @@ limitations under the License.
         min-height: 25px;
         padding-inline-start: $spacing-8;
         overflow-x: hidden;
-        overflow-y: auto;
+        overflow: auto;
         display: flex;
         flex-wrap: wrap;
 
@@ -194,6 +194,7 @@ limitations under the License.
 
     .mx_InviteDialog_addressBar {
         margin-inline-end: 0;
+        overflow: auto;
     }
 
     .mx_InviteDialog_userSections {
