diff --git a/node_modules/matrix-react-sdk/src/Avatar.ts b/node_modules/matrix-react-sdk/src/Avatar.ts
index a023ba0..13432df 100644
--- a/node_modules/matrix-react-sdk/src/Avatar.ts
+++ b/node_modules/matrix-react-sdk/src/Avatar.ts
@@ -98,6 +98,7 @@ export function defaultAvatarUrlForString(s: string): string {
     const cssVariable = `--avatar-background-colors_${colorIndex}`;
     const cssValue = getComputedStyle(document.body).getPropertyValue(cssVariable);
     const color = cssValue || defaultColors[colorIndex];
+
     let dataUrl = colorToDataURLCache.get(color);
     if (!dataUrl) {
         // validate color as this can come from account_data
