diff --git a/node_modules/matrix-react-sdk/src/IConfigOptions.ts b/node_modules/matrix-react-sdk/src/IConfigOptions.ts
index 501d8a3..19c2f3d 100644
--- a/node_modules/matrix-react-sdk/src/IConfigOptions.ts
+++ b/node_modules/matrix-react-sdk/src/IConfigOptions.ts
@@ -211,3 +211,14 @@ export interface ISsoRedirectOptions {
     immediate?: boolean;
     on_welcome_page?: boolean;
 }
+
+/* :tchap:
+* Add tchap specific options to IConfigOptions. Both interfaces get merged in compilation. https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-interfaces
+* IConfigOptions declares which options can get retrieved with SdkConfig, if not declared SdkConfig complains it does not know the parameter.
+*/
+export interface IConfigOptions {
+    tchap_features?: {
+        feature_email_notification?:[string]//activate email notification on a list of home servers, ie : "dev01.tchap.incubateur.net"
+    }
+}
+//end :tchap:
\ No newline at end of file
diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/Notifications.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/Notifications.tsx
index ec47cff..aa70cf8 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/Notifications.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/Notifications.tsx
@@ -59,6 +59,8 @@ import { SettingsSubsectionHeading } from "./shared/SettingsSubsectionHeading";
 import SettingsSubsection from "./shared/SettingsSubsection";
 import { doesRoomHaveUnreadMessages } from "../../../Unread";
 
+import TchapUIFeature from "../../../../../../src/tchap/util/TchapUIFeature";
+
 // TODO: this "view" component still has far too much application logic in it,
 // which should be factored out to other files.
 
@@ -725,7 +727,11 @@ export default class Notifications extends React.PureComponent<IProps, IState> {
                     </>
                 )}
 
+                {/* :TCHAP: show button only if feature is active on homeserver
                 {emailSwitches}
+                */}
+                { TchapUIFeature.isFeatureActiveForHomeserver("feature_email_notification") ? emailSwitches : null}
+                {/* :TCHAP: end */}
             </SettingsSubsection>
         );
     }
