diff --git a/node_modules/matrix-react-sdk/src/IConfigOptions.ts b/node_modules/matrix-react-sdk/src/IConfigOptions.ts
index 2df5e7f..daa8874 100644
--- a/node_modules/matrix-react-sdk/src/IConfigOptions.ts
+++ b/node_modules/matrix-react-sdk/src/IConfigOptions.ts
@@ -213,3 +213,14 @@ export interface ISsoRedirectOptions {
     immediate?: boolean;
     on_welcome_page?: boolean;
 }
+
+/* :tchap:
+* Add tchap specific options to IConfigOptions. Both interfaces get merged in compilation. https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-interfaces
+* IConfigOptions declares which options can get retrieved with SdkConfig, if not declared SdkConfig complains it does not know the parameter.
+*/
+export interface IConfigOptions {
+    tchap_features?: {
+        feature_email_notification?:[string]//activate email notification on a list of home servers, ie : "agent1.tchap.incubateur.net"
+    }
+}
+//end :tchap:
\ No newline at end of file
diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/Notifications.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/Notifications.tsx
index 466ecb7..f3d8e6e 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/Notifications.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/Notifications.tsx
@@ -58,6 +58,8 @@ import { Caption } from "../typography/Caption";
 import { SettingsSubsectionHeading } from "./shared/SettingsSubsectionHeading";
 import SettingsSubsection from "./shared/SettingsSubsection";
 
+import TchapUIFeature from "../../../../../../src/tchap/util/TchapUIFeature";
+
 // TODO: this "view" component still has far too much application logic in it,
 // which should be factored out to other files.
 
@@ -724,7 +726,11 @@ export default class Notifications extends React.PureComponent<IProps, IState> {
                     </>
                 )}
 
+                {/* :TCHAP: show button only if feature is active on homeserver
                 {emailSwitches}
+                */}
+                { TchapUIFeature.isFeatureActiveForHomeserver("feature_email_notification") ? emailSwitches : null}
+                {/* :TCHAP: end */}
             </SettingsSubsection>
         );
     }
