diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/SecureBackupPanel.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/SecureBackupPanel.tsx
index 2e6e6fb..3da7298 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/SecureBackupPanel.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/SecureBackupPanel.tsx
@@ -500,7 +500,9 @@ export default class SecureBackupPanel extends React.PureComponent<{}, IState> {
                     </table>
                     {extraDetails}
                 </details>
+                { /* :TCHAP: remove actions button, to block setup
                 {actionRow}
+                end :TCHAP:*/ }
             </div>
         );
     }
