diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/SecureBackupPanel.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/SecureBackupPanel.tsx
index 2e6e6fb..5571eda 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/SecureBackupPanel.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/SecureBackupPanel.tsx
@@ -30,6 +30,7 @@ import AccessibleButton from "../elements/AccessibleButton";
 import QuestionDialog from "../dialogs/QuestionDialog";
 import RestoreKeyBackupDialog from "../dialogs/security/RestoreKeyBackupDialog";
 import { accessSecretStorage } from "../../../SecurityManager";
+import TchapUIFeature from "../../../../../../src/util/TchapUIFeature"; // :TCHAP:
 
 interface IState {
     loading: boolean;
@@ -460,6 +461,12 @@ export default class SecureBackupPanel extends React.PureComponent<{}, IState> {
         let actionRow;
         if (actions.length) {
             actionRow = <div className="mx_SecureBackupPanel_buttonRow">{actions}</div>;
+            // :TCHAP: hide action buttons if secure storage is not supported
+            // actionRow = <div className="mx_SecureBackupPanel_buttonRow">{actions}</div>;
+            if (!TchapUIFeature.isCrossSigningAndSecureStorageActive) {
+                actionRow = null;
+            }
+            // end :TCHAP:
         }
 
         return (
