diff --git a/node_modules/matrix-react-sdk/src/stores/RoomViewStore.tsx b/node_modules/matrix-react-sdk/src/stores/RoomViewStore.tsx
index 30f01bd..3c1c49b 100644
--- a/node_modules/matrix-react-sdk/src/stores/RoomViewStore.tsx
+++ b/node_modules/matrix-react-sdk/src/stores/RoomViewStore.tsx
@@ -603,6 +603,12 @@ export class RoomViewStore extends EventEmitter {
         let description: ReactNode = err.message ? err.message : JSON.stringify(err);
         logger.log("Failed to join room:", description);
 
+        /* :TCHAP: add this for a translation */
+        if (typeof description === 'string' && description.includes('You are not invited to this room')) {
+            description = _t("Access possible only by invitation of a member of the room.");
+        }
+        /* end :TCHAP: */
+
         if (err.name === "ConnectionError") {
             description = _t("There was an error joining.");
         } else if (err.errcode === "M_INCOMPATIBLE_ROOM_VERSION") {
