diff --git a/node_modules/matrix-react-sdk/src/components/views/context_menus/MessageContextMenu.tsx b/node_modules/matrix-react-sdk/src/components/views/context_menus/MessageContextMenu.tsx
index ad56409..fa0407d 100644
--- a/node_modules/matrix-react-sdk/src/components/views/context_menus/MessageContextMenu.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/context_menus/MessageContextMenu.tsx
@@ -678,7 +678,9 @@ export default class MessageContextMenu extends React.Component<IProps, IState>
                 <IconizedContextMenuOptionList>
                     {reactButton}
                     {replyButton}
+                    {/* TCHAP : Remove Reply In Thread Button 
                     {replyInThreadButton}
+                    TCHAP */}
                     {editButton}
                 </IconizedContextMenuOptionList>
             );
diff --git a/node_modules/matrix-react-sdk/src/components/views/right_panel/RoomHeaderButtons.tsx b/node_modules/matrix-react-sdk/src/components/views/right_panel/RoomHeaderButtons.tsx
index 4d7e873..1e502b4 100644
--- a/node_modules/matrix-react-sdk/src/components/views/right_panel/RoomHeaderButtons.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/right_panel/RoomHeaderButtons.tsx
@@ -292,6 +292,7 @@ export default class RoomHeaderButtons extends HeaderButtons<IProps> {
                 onClick={this.onTimelineCardClicked}
             />,
         );
+        /* Tchap Remove Thread from Header Buttons 
         rightPanelPhaseButtons.set(
             RightPanelPhases.ThreadPanel,
             <HeaderButton
@@ -306,6 +307,7 @@ export default class RoomHeaderButtons extends HeaderButtons<IProps> {
                 <UnreadIndicator color={this.state.threadNotificationColor} />
             </HeaderButton>,
         );
+        TCHAP Remove Thread from Header Buttons */
         rightPanelPhaseButtons.set(
             RightPanelPhases.NotificationPanel,
             <HeaderButton
