diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/CryptographyPanel.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/CryptographyPanel.tsx
index 34b52e4..8085f20 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/CryptographyPanel.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/CryptographyPanel.tsx
@@ -72,7 +72,9 @@ export default class CryptographyPanel extends React.Component<IProps, IState> {
         }
 
         return (
-            <div className="mx_SettingsTab_section mx_CryptographyPanel">
+            <div className="mx_CryptographyPanel">
+            {/* :TCHAP: remove class <div className="mx_SettingsTab_section mx_CryptographyPanel"> */}
+                {/* :TCHAP: modify content
                 <span className="mx_SettingsTab_subheading">{_t("Cryptography")}</span>
                 <table className="mx_SettingsTab_subsectionText mx_CryptographyPanel_sessionInfo">
                     <tbody>
@@ -92,7 +94,36 @@ export default class CryptographyPanel extends React.Component<IProps, IState> {
                         </tr>
                     </tbody>
                 </table>
-                {importExportButtons}
+                {importExportButtons} */}
+
+                <span className="mx_SettingsTab_subheading">{_t("Tchap keys")}</span>
+                <div className="mx_SettingsTab_subsectionText">{_t("These keys only apply to the current session.")}</div>
+                <div className="mx_SettingsTab_subsectionText"><b>{_t("Please note this is not your recovery code for your automatic backup.")}</b></div>
+                
+                <details className="mx_SettingsTab_subsectionText">
+                    <summary>{_t("Advanced")}</summary>
+                    <table className="mx_CryptographyPanel_sessionInfo">
+                        <tbody>
+                            <tr>
+                                <td>{_t("Session ID:")}</td>
+                                <td>
+                                    <code>{deviceId}</code>
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>{_t("Tchap keys:")}</td>
+                                <td>
+                                    <code>
+                                        <b>{identityKey}</b>
+                                    </code>
+                                </td>
+                            </tr>
+                        </tbody>
+                    </table>
+                </details>
+
+                <div className="mx_CrossSigningPanel_buttonRow">{importExportButtons}</div>
+                {/* end :TCHAP: */}
                 {noSendUnverifiedSetting}
             </div>
         );
diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SecurityUserSettingsTab.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SecurityUserSettingsTab.tsx
index 9fe4528..df362fa 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SecurityUserSettingsTab.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SecurityUserSettingsTab.tsx
@@ -292,7 +292,7 @@ export default class SecurityUserSettingsTab extends React.Component<IProps, ISt
     public render(): React.ReactNode {
         const secureBackup = (
             <div className="mx_SettingsTab_section">
-                <span className="mx_SettingsTab_subheading">{_t("Secure Backup")}</span>
+                <span className="mx_SettingsTab_heading">{_t("Secure Backup")}</span>
                 <div className="mx_SettingsTab_subsectionText">
                     <SecureBackupPanel />
                 </div>
@@ -423,14 +423,16 @@ export default class SecurityUserSettingsTab extends React.Component<IProps, ISt
         return (
             <div className="mx_SettingsTab mx_SecurityUserSettingsTab">
                 {warning}
-                {devicesSection}
-                <div className="mx_SettingsTab_heading">{_t("Encryption")}</div>
-                <div className="mx_SettingsTab_section">
+                {/* :TCHAP: change position {devicesSection} */}
+                {/* :TCHAP: remove class <div className="mx_SettingsTab_section"> */}
+                <div>
                     {secureBackup}
-                    {eventIndex}
-                    {crossSigning}
-                    <CryptographyPanel />
                 </div>
+                {/* {eventIndex} :TCHAP: remove section */}
+                {devicesSection}
+                <div className="mx_SettingsTab_heading">{_t("Encryption")}</div>
+                {crossSigning}
+                <CryptographyPanel />
                 {privacySection}
                 {advancedSection}
             </div>
