diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/CryptographyPanel.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/CryptographyPanel.tsx
index 13151d2..6814313 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/CryptographyPanel.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/CryptographyPanel.tsx
@@ -72,6 +72,7 @@ export default class CryptographyPanel extends React.Component<IProps, IState> {
             );
         }
 
+        /* :TCHAP: redesign the returned component
         return (
             <SettingsSubsection heading={_t("settings|security|cryptography_section")}>
                 <SettingsSubsectionText>
@@ -96,7 +97,44 @@ export default class CryptographyPanel extends React.Component<IProps, IState> {
                 {noSendUnverifiedSetting}
             </SettingsSubsection>
         );
+        */
+        return (
+            <SettingsSubsection heading={_t("settings|security|cryptography_section")}> {/* :TCHAP: name changed in translations */}
+            <div className="mx_SettingsTab_subsectionText">
+                <div>
+                    {_t("These keys only apply to the current session.")}
+                </div>
+                <div>
+                    <b>{_t("Please note this is not your recovery code for your automatic backup.")}</b>
+                </div>
+            </div>
+            <details>
+                <summary>{_t("common|advanced")}</summary>
+                <SettingsSubsectionText>
+                    <table className="mx_CryptographyPanel_sessionInfo">
+                        <tr>
+                            <th scope="row">{_t("settings|security|session_id")}</th>
+                            <td>
+                                <code>{deviceId}</code>
+                            </td>
+                        </tr>
+                        <tr>
+                            <th scope="row">{_t("settings|security|session_key")}</th>
+                            <td>
+                                <code>
+                                    <b>{identityKey}</b>
+                                </code>
+                            </td>
+                        </tr>
+                    </table>
+                </SettingsSubsectionText>
+            </details>
+            {importExportButtons}
+            {noSendUnverifiedSetting}
+            </SettingsSubsection>
+        );
     }
+    // end :TCHAP:
 
     private onExportE2eKeysClicked = (): void => {
         Modal.createDialogAsync(
diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SecurityUserSettingsTab.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SecurityUserSettingsTab.tsx
index 74511df..5b124a1 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SecurityUserSettingsTab.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SecurityUserSettingsTab.tsx
@@ -275,11 +275,20 @@ export default class SecurityUserSettingsTab extends React.Component<IProps, ISt
     }
 
     public render(): React.ReactNode {
+        // :TCHAP: we're moving this subsection out, so it's now a section
+        const secureBackup = (
+            <SettingsSection heading={_t("common|secure_backup")}>
+                <SecureBackupPanel />
+            </SettingsSection>
+        );
+        /*
         const secureBackup = (
             <SettingsSubsection heading={_t("common|secure_backup")}>
                 <SecureBackupPanel />
             </SettingsSubsection>
         );
+        */
+        // end :TCHAP:
 
         const eventIndex = (
             <SettingsSubsection heading={_t("settings|security|message_search_section")}>
@@ -354,13 +363,14 @@ export default class SecurityUserSettingsTab extends React.Component<IProps, ISt
         return (
             <SettingsTab>
                 {warning}
+                { /* :TCHAP: move secureBackup and privacySection, and remove eventIndex */ }
+                {secureBackup}
                 <SettingsSection heading={_t("settings|security|encryption_section")}>
-                    {secureBackup}
-                    {eventIndex}
                     {crossSigning}
                     <CryptographyPanel />
+                    {privacySection}
                 </SettingsSection>
-                {privacySection}
+                { /* end :TCHAP: */ }
                 {advancedSection}
             </SettingsTab>
         );
