diff --git a/node_modules/matrix-js-sdk/src/interactive-auth.ts b/node_modules/matrix-js-sdk/src/interactive-auth.ts
index 47ff4cd..9172a50 100644
--- a/node_modules/matrix-js-sdk/src/interactive-auth.ts
+++ b/node_modules/matrix-js-sdk/src/interactive-auth.ts
@@ -634,8 +634,17 @@ export class InteractiveAuth<T> {
 
     // Returns a low number for flows we consider best. Counts increase for longer flows and even more so
     // for flows which contain stages not listed in `supportedStages`.
-    private scoreFlow(flow: UIAFlow): number {
+    private scoreFlow(flow: UIAFlow, preferSSO: boolean): number {
         let score = flow.stages.length;
+        
+        // :TCHAP: prefrer-sso-when-multiple-choice
+        // Prefer sso flow over password by giving password a higher score
+        // If we have password and sso flow at the same time, we will choose sso flow
+        if (flow.stages.includes(AuthType.Password) && preferSSO) {
+            score += 2;
+        }
+        // end :TCHAP: 
+
         if (this.supportedStages !== undefined) {
             // Add 10 points to the score for each unsupported stage in the flow.
             score += flow.stages.filter((stage) => !this.supportedStages!.has(stage)).length * 10;
@@ -665,9 +674,16 @@ export class InteractiveAuth<T> {
         const haveEmail = Boolean(this.inputs.emailAddress) || Boolean(this.emailSid);
         const haveMsisdn = Boolean(this.inputs.phoneCountry) && Boolean(this.inputs.phoneNumber);
 
+        // :TCHAP: prefer-sso-when-multiple-choice only if  password and sso flow are present
+        const hasPasswordFlow = flows.some(flow => flow.stages.includes(AuthType.Password));
+        const hasSsoFlow = flows.some(flow => flow.stages.includes(AuthType.Sso) || flow.stages.includes(AuthType.SsoUnstable));
+        
+        const preferSSO = hasPasswordFlow && hasSsoFlow;
+        // end :TCHAP:
+
         // Flows are not represented in a significant order, so we can choose any we support best
         // Sort flows based on how many unsupported stages they contain ascending
-        flows.sort((a, b) => this.scoreFlow(a) - this.scoreFlow(b));
+        flows.sort((a, b) => this.scoreFlow(a, preferSSO) - this.scoreFlow(b, preferSSO));
 
         for (const flow of flows) {
             let flowHasEmail = false;
