diff --git a/node_modules/matrix-react-sdk/src/components/views/dialogs/BugReportDialog.tsx b/node_modules/matrix-react-sdk/src/components/views/dialogs/BugReportDialog.tsx
index 4074db6..91271bb 100644
--- a/node_modules/matrix-react-sdk/src/components/views/dialogs/BugReportDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/dialogs/BugReportDialog.tsx
@@ -231,7 +231,7 @@ export default class BugReportDialog extends React.Component<IProps, IState> {
                                     a: (sub) => (
                                         <a
                                             target="_blank"
-                                            href="https://github.com/vector-im/element-web/issues/new/choose"
+                                            href="https://github.com/tchapgouv/tchap-web-v4/issues/new/choose"
                                         >
                                             {sub}
                                         </a>
diff --git a/node_modules/matrix-react-sdk/src/components/views/dialogs/FeedbackDialog.tsx b/node_modules/matrix-react-sdk/src/components/views/dialogs/FeedbackDialog.tsx
index 9996e65..548ff1a 100644
--- a/node_modules/matrix-react-sdk/src/components/views/dialogs/FeedbackDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/dialogs/FeedbackDialog.tsx
@@ -29,8 +29,8 @@ import { useStateToggle } from "../../../hooks/useStateToggle";
 import StyledCheckbox from "../elements/StyledCheckbox";
 
 const existingIssuesUrl =
-    "https://github.com/vector-im/element-web/issues" + "?q=is%3Aopen+is%3Aissue+sort%3Areactions-%2B1-desc";
-const newIssueUrl = "https://github.com/vector-im/element-web/issues/new/choose";
+    "https://github.com/tchapgouv/tchap-web-v4/issues" + "?q=is%3Aopen+is%3Aissue+sort%3Areactions-%2B1-desc";
+const newIssueUrl = "https://github.com/tchapgouv/tchap-web-v4/issues/new/choose";
 
 interface IProps {
     feature?: string;
diff --git a/node_modules/matrix-react-sdk/src/components/views/elements/ErrorBoundary.tsx b/node_modules/matrix-react-sdk/src/components/views/elements/ErrorBoundary.tsx
index 7603041..3b9833f 100644
--- a/node_modules/matrix-react-sdk/src/components/views/elements/ErrorBoundary.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/elements/ErrorBoundary.tsx
@@ -75,7 +75,7 @@ export default class ErrorBoundary extends React.PureComponent<{}, IState> {
 
     public render(): ReactNode {
         if (this.state.error) {
-            const newIssueUrl = "https://github.com/vector-im/element-web/issues/new/choose";
+            const newIssueUrl = "https://github.com/tchapgouv/tchap-web-v4/issues/new/choose";
 
             let bugReportSection;
             if (SdkConfig.get().bug_report_endpoint_url) {
diff --git a/node_modules/matrix-react-sdk/src/components/views/rooms/RoomPreviewBar.tsx b/node_modules/matrix-react-sdk/src/components/views/rooms/RoomPreviewBar.tsx
index a968bd3..709a715 100644
--- a/node_modules/matrix-react-sdk/src/components/views/rooms/RoomPreviewBar.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/rooms/RoomPreviewBar.tsx
@@ -564,7 +564,7 @@ export default class RoomPreviewBar extends React.Component<IProps, IState> {
                         {
                             issueLink: (label) => (
                                 <a
-                                    href="https://github.com/vector-im/element-web/issues/new/choose"
+                                    href="https://github.com/tchapgouv/tchap-web-v4/issues/new/choose"
                                     target="_blank"
                                     rel="noreferrer noopener"
                                 >
