diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SecurityUserSettingsTab.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SecurityUserSettingsTab.tsx
index 177c3b4..a86b6b1 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SecurityUserSettingsTab.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/SecurityUserSettingsTab.tsx
@@ -252,14 +252,9 @@ export default class SecurityUserSettingsTab extends React.Component<IProps, ISt
     }
 
     public render(): JSX.Element {
-        const secureBackup = (
-            <div className='mx_SettingsTab_section'>
-                <span className="mx_SettingsTab_subheading">{ _t("Secure Backup") }</span>
-                <div className='mx_SettingsTab_subsectionText'>
-                    <SecureBackupPanel />
-                </div>
-            </div>
-        );
+        // :TCHAP: hide secure backup
+        const secureBackup = null;
+        // end :TCHAP: 
 
         const eventIndex = (
             <div className="mx_SettingsTab_section">
