diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/room/GeneralRoomSettingsTab.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/room/GeneralRoomSettingsTab.tsx
index 8dc361b..3a5e10d 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/room/GeneralRoomSettingsTab.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/room/GeneralRoomSettingsTab.tsx
@@ -89,6 +89,7 @@ export default class GeneralRoomSettingsTab extends React.Component<IProps, ISta
                     <RoomProfileSettings roomId={room.roomId} />
                 </SettingsSection>
 
+                {/* :TCHAP: no aliases for rooms
                 <SettingsSection heading={_t("Room Addresses")}>
                     <AliasSettings
                         roomId={room.roomId}
@@ -97,6 +98,7 @@ export default class GeneralRoomSettingsTab extends React.Component<IProps, ISta
                         canonicalAliasEvent={canonicalAliasEv}
                     />
                 </SettingsSection>
+                end :TCHAP: */}
 
                 <SettingsSection heading={_t("Other")}>
                     {urlPreviewSettings}
