diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/room/GeneralRoomSettingsTab.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/room/GeneralRoomSettingsTab.tsx
index 21258bf..4f9adb4 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/room/GeneralRoomSettingsTab.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/room/GeneralRoomSettingsTab.tsx
@@ -89,6 +89,7 @@ export default class GeneralRoomSettingsTab extends React.Component<IProps, ISta
                     <RoomProfileSettings roomId={this.props.roomId} />
                 </div>
 
+                {/* :TCHAP: no aliases for rooms
                 <div className="mx_SettingsTab_heading">{_t("Room Addresses")}</div>
                 <AliasSettings
                     roomId={this.props.roomId}
@@ -96,6 +97,7 @@ export default class GeneralRoomSettingsTab extends React.Component<IProps, ISta
                     canSetAliases={canSetAliases}
                     canonicalAliasEvent={canonicalAliasEv}
                 />
+                end :TCHAP: */}
                 <div className="mx_SettingsTab_heading">{_t("Other")}</div>
                 {urlPreviewSettings}
                 {leaveSection}
