diff --git a/node_modules/matrix-react-sdk/res/css/structures/_UserMenu.pcss b/node_modules/matrix-react-sdk/res/css/structures/_UserMenu.pcss
index 9c9e1ce..6fe1939 100644
--- a/node_modules/matrix-react-sdk/res/css/structures/_UserMenu.pcss
+++ b/node_modules/matrix-react-sdk/res/css/structures/_UserMenu.pcss
@@ -198,6 +198,10 @@ limitations under the License.
         mask-image: url("$(res)/img/element-icons/settings.svg");
     }
 
+    .mx_UserMenu_iconInfo::before {
+        mask-image: url("$(res)/img/element-icons/info.svg");
+    }
+
     .mx_UserMenu_iconMessage::before {
         mask-image: url("$(res)/img/element-icons/feedback.svg");
     }
diff --git a/node_modules/matrix-react-sdk/src/components/structures/UserMenu.tsx b/node_modules/matrix-react-sdk/src/components/structures/UserMenu.tsx
index 5d2ba89..76f4a34 100644
--- a/node_modules/matrix-react-sdk/src/components/structures/UserMenu.tsx
+++ b/node_modules/matrix-react-sdk/src/components/structures/UserMenu.tsx
@@ -246,6 +246,10 @@ export default class UserMenu extends React.Component<IProps, IState> {
         this.setState({ contextMenuPosition: null }); // also close the menu
     };
 
+    private onOpenFAQPage = () => {
+        window.open("https://www.tchap.gouv.fr/faq", '_blank');
+    };
+
     private onProvideFeedback = (ev: ButtonEvent): void => {
         ev.preventDefault();
         ev.stopPropagation();
@@ -363,6 +367,11 @@ export default class UserMenu extends React.Component<IProps, IState> {
                     onClick={(e) => this.onSettingsOpen(e)}
                 />
                 {feedbackButton}
+                <IconizedContextMenuOption
+                    iconClassName="mx_UserMenu_iconInfo"
+                    label={_t("Help")}
+                    onClick={this.onOpenFAQPage}
+                />
                 <IconizedContextMenuOption
                     className="mx_IconizedContextMenu_option_red"
                     iconClassName="mx_UserMenu_iconSignOut"
