diff --git a/node_modules/matrix-react-sdk/src/components/structures/auth/SessionLockStolenView.tsx b/node_modules/matrix-react-sdk/src/components/structures/auth/SessionLockStolenView.tsx
index 0c0bc01..1a5ef24 100644
--- a/node_modules/matrix-react-sdk/src/components/structures/auth/SessionLockStolenView.tsx
+++ b/node_modules/matrix-react-sdk/src/components/structures/auth/SessionLockStolenView.tsx
@@ -28,8 +28,13 @@ export function SessionLockStolenView(): JSX.Element {
 
     return (
         <SplashPage className="mx_SessionLockStolenView">
+            {/** :TCHAP: better messaging
             <h1>{_t("common|error")}</h1>
             <h2>{_t("%(brand)s has been opened in another tab.", { brand })}</h2>
+            */}
+            <h1>{_t("%(brand)s has been connected in another tab", { brand })}</h1>
+            <h2>{_t("You can close this disconnected tab, and go to the other %(brand)s tab.", { brand })}</h2>
+            {/** end :TCHAP: */}
         </SplashPage>
     );
 }
