diff --git a/node_modules/matrix-react-sdk/src/components/structures/auth/SessionLockStolenView.tsx b/node_modules/matrix-react-sdk/src/components/structures/auth/SessionLockStolenView.tsx
index de93624..64378b8 100644
--- a/node_modules/matrix-react-sdk/src/components/structures/auth/SessionLockStolenView.tsx
+++ b/node_modules/matrix-react-sdk/src/components/structures/auth/SessionLockStolenView.tsx
@@ -28,8 +28,13 @@ export function SessionLockStolenView(): JSX.Element {
 
     return (
         <SplashPage className="mx_SessionLockStolenView">
+            {/** :TCHAP: better messaging
             <h1>{_t("common|error")}</h1>
             <h2>{_t("error_app_open_in_another_tab", { brand })}</h2>
-        </SplashPage>
-    );
+            */}
+            <h1>{_t("error_app_open_in_another_tab", { brand })}</h1>
+            <h2>{_t("You can close this disconnected tab, and go to the other %(brand)s tab.", { brand })}</h2>
+            {/** end :TCHAP: */}
+       </SplashPage>
+   );
 }
