diff --git a/node_modules/matrix-react-sdk/src/components/views/dialogs/IncomingSasDialog.tsx b/node_modules/matrix-react-sdk/src/components/views/dialogs/IncomingSasDialog.tsx
index a562760..2d13606 100644
--- a/node_modules/matrix-react-sdk/src/components/views/dialogs/IncomingSasDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/dialogs/IncomingSasDialog.tsx
@@ -175,6 +175,7 @@ export default class IncomingSasDialog extends React.Component<IProps, IState> {
             profile = <Spinner />;
         }
 
+        /* :TCHAP: remove code
         const userDetailText = [
             <p key="p1">{_t("encryption|verification|incoming_sas_user_dialog_text_1")}</p>,
             <p key="p2">
@@ -190,11 +191,23 @@ export default class IncomingSasDialog extends React.Component<IProps, IState> {
             <p key="p1">{_t("encryption|verification|incoming_sas_device_dialog_text_1")}</p>,
             <p key="p2">{_t("encryption|verification|incoming_sas_device_dialog_text_2")}</p>,
         ];
+        end :TCHAP: */
 
         return (
             <div>
+                {/* :TCHAP: remove code
                 {profile}
                 {isSelf ? selfDetailText : userDetailText}
+                end :TCHAP: */}
+
+                {/* :TCHAP: simplify modal message */}
+                <p>{ _t(
+                    "One of your devices <b>wants to check your Tchap Keys</b> to unlock your messages.",
+                    {},
+                    { b: sub => <b>{ sub }</b> },
+                ) }</p>
+                {/* end :TCHAP: */}
+
                 <DialogButtons
                     primaryButton={_t("action|continue")}
                     hasCancel={true}
diff --git a/node_modules/matrix-react-sdk/src/components/views/dialogs/VerificationRequestDialog.tsx b/node_modules/matrix-react-sdk/src/components/views/dialogs/VerificationRequestDialog.tsx
index c6a86ae..0c3f80a 100644
--- a/node_modules/matrix-react-sdk/src/components/views/dialogs/VerificationRequestDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/dialogs/VerificationRequestDialog.tsx
@@ -49,9 +49,12 @@ export default class VerificationRequestDialog extends React.Component<IProps, I
         const request = this.state.verificationRequest;
         const otherUserId = request?.otherUserId;
         const member = this.props.member || (otherUserId ? MatrixClientPeg.safeGet().getUser(otherUserId) : null);
+        /* :TCHAP: remove code
         const title = request?.isSelfVerification
             ? _t("encryption|verification|verification_dialog_title_device")
             : _t("encryption|verification|verification_dialog_title_user");
+        end :TCHAP: */
+        const title = _t("Incoming Verification Request"); // :TCHAP:
 
         if (!member) return null;
 
diff --git a/node_modules/matrix-react-sdk/src/components/views/right_panel/VerificationPanel.tsx b/node_modules/matrix-react-sdk/src/components/views/right_panel/VerificationPanel.tsx
index c087219..e626a42 100644
--- a/node_modules/matrix-react-sdk/src/components/views/right_panel/VerificationPanel.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/right_panel/VerificationPanel.tsx
@@ -295,10 +295,13 @@ export default class VerificationPanel extends React.PureComponent<IProps, IStat
                 logger.warn("Verified device we don't know about: " + this.props.request.otherDeviceId);
                 description = _t("encryption|verification|successful_own_device");
             } else {
+                description = _t("The sharing of your Tchap Keys has succeeded. Your messages will be unlocked."); // :TCHAP:
+                /* :TCHAP:
                 description = _t("encryption|verification|successful_device", {
                     deviceName: device.displayName,
                     deviceId: device.deviceId,
                 });
+                end :TCHAP: */
             }
         } else {
             description = _t("encryption|verification|successful_user", {
diff --git a/node_modules/matrix-react-sdk/src/components/views/verification/VerificationComplete.tsx b/node_modules/matrix-react-sdk/src/components/views/verification/VerificationComplete.tsx
index 002bac7..3aa6309 100644
--- a/node_modules/matrix-react-sdk/src/components/views/verification/VerificationComplete.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/verification/VerificationComplete.tsx
@@ -27,9 +27,14 @@ export default class VerificationComplete extends React.Component<IProps> {
     public render(): React.ReactNode {
         return (
             <div>
+                { /** :TCHAP: replace title and description */}
+                <h2>{ _t("Incoming Verification Request") }</h2>
+                <p>{ _t("The sharing of your Tchap Keys has succeeded. Your messages will be unlocked.") }</p>
+                {/*
                 <h2>{_t("encryption|verification|complete_title")}</h2>
                 <p>{_t("encryption|verification|complete_description")}</p>
                 <p>{_t("encryption|verification|explainer")}</p>
+                end :TCHAP: */}
                 <DialogButtons
                     onPrimaryButtonClick={this.props.onDone}
                     primaryButton={_t("encryption|verification|complete_action")}
