diff --git a/node_modules/matrix-js-sdk/src/http-api/fetch.ts b/node_modules/matrix-js-sdk/src/http-api/fetch.ts
index 38d9b4a..d4bc0a0 100644
--- a/node_modules/matrix-js-sdk/src/http-api/fetch.ts
+++ b/node_modules/matrix-js-sdk/src/http-api/fetch.ts
@@ -160,6 +160,12 @@ export class FetchHttpApi<O extends IHttpOpts> {
             } else if (err.errcode == "M_CONSENT_NOT_GIVEN") {
                 this.eventEmitter.emit(HttpApiEvent.NoConsent, err.message, err.data.consent_uri);
             }
+            //:tchap: this error is sent by the synapse module (synapse-email-account-validity)[https://github.com/matrix-org/synapse-email-account-validity] 
+            // when user account has expired
+            else if (err.errcode == 'ORG_MATRIX_EXPIRED_ACCOUNT') {
+                this.eventEmitter.emit(HttpApiEvent.ORG_MATRIX_EXPIRED_ACCOUNT, err);
+            }
+            //:tchap end
         });
 
         // return the original promise, otherwise tests break due to it having to
diff --git a/node_modules/matrix-js-sdk/src/http-api/interface.ts b/node_modules/matrix-js-sdk/src/http-api/interface.ts
index 57e8a18..525860d 100644
--- a/node_modules/matrix-js-sdk/src/http-api/interface.ts
+++ b/node_modules/matrix-js-sdk/src/http-api/interface.ts
@@ -75,6 +75,10 @@ export interface IContentUri {
 export enum HttpApiEvent {
     SessionLoggedOut = "Session.logged_out",
     NoConsent = "no_consent",
+    //:tchap: this error is sent by the synapse module (synapse-email-account-validity)[https://github.com/matrix-org/synapse-email-account-validity] 
+    //when user account has expired
+    ORG_MATRIX_EXPIRED_ACCOUNT = "ORG_MATRIX_EXPIRED_ACCOUNT",
+    //:tchap end
 }
 
 export type HttpApiEventHandlerMap = {
@@ -102,6 +106,9 @@ export type HttpApiEventHandlerMap = {
      * ```
      */
     [HttpApiEvent.NoConsent]: (message: string, consentUri: string) => void;
+    //:tchap: this error is sent by the synapse module (synapse-email-account-validity)[https://github.com/matrix-org/synapse-email-account-validity] 
+    //when user account has expired
+    [HttpApiEvent.ORG_MATRIX_EXPIRED_ACCOUNT]: (err: MatrixError) => void;
 };
 
 export interface UploadProgress {
diff --git a/node_modules/matrix-js-sdk/src/sync.ts b/node_modules/matrix-js-sdk/src/sync.ts
index f2ec1b3..c9266b7 100644
--- a/node_modules/matrix-js-sdk/src/sync.ts
+++ b/node_modules/matrix-js-sdk/src/sync.ts
@@ -614,6 +614,14 @@ export class SyncApi {
             this.updateSyncState(SyncState.Error, { error });
             return true;
         }
+        //:tchap: this error is sent by (synapse-email-account-validity)[https://github.com/matrix-org/synapse-email-account-validity] 
+        // when user account has expired. client must be stopped and sync aborted when the user account is not valid anymore
+        if (error.errcode === "ORG_MATRIX_EXPIRED_ACCOUNT") {
+            // The account is expired
+            this.stop();
+            this.updateSyncState(SyncState.Error, { error });
+            return true;
+        }
         return false;
     }
 
