diff --git a/node_modules/matrix-react-sdk/src/components/views/rooms/DecryptionFailureBar.tsx b/node_modules/matrix-react-sdk/src/components/views/rooms/DecryptionFailureBar.tsx
index 7849f63..b393746 100644
--- a/node_modules/matrix-react-sdk/src/components/views/rooms/DecryptionFailureBar.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/rooms/DecryptionFailureBar.tsx
@@ -180,27 +180,37 @@ export const DecryptionFailureBar: React.FC<IProps> = ({ failures }) => {
                     )}
                 </React.Fragment>
             );
+            /* :tchap: replace reset button by open-security button
             button = (
                 <AccessibleButton kind="primary" onClick={onResetClick}>
                     {_t("Reset")}
                 </AccessibleButton>
             );
+             */
+            button = (
+                <AccessibleButton kind="primary" onClick={onDeviceListClick}>
+                    {_t("Security & Privacy")}
+                </AccessibleButton>
+            );
+            /* :tchap: end */ 
         }
     } else if (hasOtherVerifiedDevices) {
         headline = <React.Fragment>{_t("Open another device to load encrypted messages")}</React.Fragment>;
-        body = (
-            <React.Fragment>
-                {_t(
-                    "This device is requesting decryption keys from your other devices. " +
-                        "Opening one of your other devices may speed this up.",
-                )}
-            </React.Fragment>
-        );
-        button = (
-            <AccessibleButton kind="primary_outline" onClick={onDeviceListClick}>
-                {_t("View your device list")}
-            </AccessibleButton>
-        );
+        // :TCHAP: shrink long banner
+        // body = (
+        //     <React.Fragment>
+        //         {_t(
+        //             "This device is requesting decryption keys from your other devices. " +
+        //                 "Opening one of your other devices may speed this up.",
+        //         )}
+        //     </React.Fragment>
+        // );
+        // button = (
+        //     <AccessibleButton kind="primary_outline" onClick={onDeviceListClick}>
+        //         {_t("View your device list")}
+        //     </AccessibleButton>
+        // );
+        // end :TCHAP:
     } else {
         headline = <React.Fragment>{_t("Some messages could not be decrypted")}</React.Fragment>;
         body = (
