diff --git a/node_modules/matrix-js-sdk/src/crypto/index.ts b/node_modules/matrix-js-sdk/src/crypto/index.ts
index 7c0092b..9d0358b 100644
--- a/node_modules/matrix-js-sdk/src/crypto/index.ts
+++ b/node_modules/matrix-js-sdk/src/crypto/index.ts
@@ -2269,6 +2269,12 @@ export class Crypto extends TypedEventEmitter<CryptoEvent, CryptoEventHandlerMap
                 await upload({ shouldEmit: true });
                 // XXX: we'll need to wait for the device list to be updated
             }
+
+            //:tchap: remove this patch after release of element with
+            // https://github.com/matrix-org/matrix-js-sdk/pull/2921
+            // redo key requests after verification
+            this.cancelAndResendAllOutgoingKeyRequests();
+            //:tchap: end
         }
 
         const deviceObj = DeviceInfo.fromStorage(dev, deviceId);
