diff --git a/node_modules/matrix-react-sdk/src/components/views/dialogs/security/AccessSecretStorageDialog.tsx b/node_modules/matrix-react-sdk/src/components/views/dialogs/security/AccessSecretStorageDialog.tsx
index a261e44..046b574 100644
--- a/node_modules/matrix-react-sdk/src/components/views/dialogs/security/AccessSecretStorageDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/dialogs/security/AccessSecretStorageDialog.tsx
@@ -48,6 +48,7 @@ interface IProps {
 }
 
 interface IState {
+    displayPassword: boolean;
     recoveryKey: string;
     recoveryKeyValid: boolean | null;
     recoveryKeyCorrect: boolean | null;
@@ -69,6 +70,7 @@ export default class AccessSecretStorageDialog extends React.PureComponent<IProp
         super(props);
 
         this.state = {
+            displayPassword: false,
             recoveryKey: "",
             recoveryKeyValid: null,
             recoveryKeyCorrect: null,
@@ -80,6 +82,16 @@ export default class AccessSecretStorageDialog extends React.PureComponent<IProp
         };
     }
 
+    // :TCHAP: add functionality that displays the Recovery Code for 2mn when clicking on the input eye icon 
+    private setDisplayPassword = (): void => {
+        this.setState({ displayPassword: true });
+    
+        setTimeout(() => {
+          this.setState({ displayPassword: false });
+        }, 60 * 2000);
+    };
+    // end :TCHAP:
+
     private onCancel = (): void => {
         if (this.state.resetting) {
             this.setState({ resetting: false });
@@ -403,7 +415,7 @@ export default class AccessSecretStorageDialog extends React.PureComponent<IProp
                         <div className="mx_AccessSecretStorageDialog_recoveryKeyEntry">
                             <div className="mx_AccessSecretStorageDialog_recoveryKeyEntry_textInput">
                                 <Field
-                                    type="password"
+                                    type={this.state.displayPassword ? "text": "password"}
                                     id="mx_securityKey"
                                     label={_t("Security Key")}
                                     value={this.state.recoveryKey}
@@ -411,15 +423,30 @@ export default class AccessSecretStorageDialog extends React.PureComponent<IProp
                                     autoFocus={true}
                                     forceValidity={this.state.recoveryKeyCorrect ?? undefined}
                                     autoComplete="off"
+                                    postfixComponent={(
+                                        <div
+                                            className="tc_textInput_postfixComponent"
+                                            onClick={() => this.setDisplayPassword()}
+                                        >
+                                            <img
+                                                src={require("../../../../../../../res/img/grey-eye.svg").default}
+                                                width="24"
+                                                height="24"
+                                                alt={_t("Eye")}
+                                            />
+                                        </div>
+                                    )}
                                 />
                             </div>
+                            {/*
+                             :TCHAP: hide for csss feature
                             <span className="mx_AccessSecretStorageDialog_recoveryKeyEntry_entryControlSeparatorText">
                                 {_t("%(securityKey)s or %(recoveryFile)s", {
                                     recoveryFile: "",
                                     securityKey: "",
                                 })}
                             </span>
-                            <div>
+                             <div>
                                 <input
                                     type="file"
                                     className="mx_AccessSecretStorageDialog_recoveryKeyEntry_fileInput"
@@ -430,7 +457,8 @@ export default class AccessSecretStorageDialog extends React.PureComponent<IProp
                                 <AccessibleButton kind="primary" onClick={this.onRecoveryKeyFileUploadClick}>
                                     {_t("Upload")}
                                 </AccessibleButton>
-                            </div>
+                            :end TCHAP
+                            </div> */}
                         </div>
                         {recoveryKeyFeedback}
                         <DialogButtons
