diff --git a/node_modules/matrix-react-sdk/src/components/views/dialogs/security/AccessSecretStorageDialog.tsx b/node_modules/matrix-react-sdk/src/components/views/dialogs/security/AccessSecretStorageDialog.tsx
index e6a5c67..600c0fd 100644
--- a/node_modules/matrix-react-sdk/src/components/views/dialogs/security/AccessSecretStorageDialog.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/dialogs/security/AccessSecretStorageDialog.tsx
@@ -407,13 +407,15 @@ export default class AccessSecretStorageDialog extends React.PureComponent<IProp
                                     autoComplete="off"
                                 />
                             </div>
+                            {/*
+                             :TCHAP: hide for csss feature
                             <span className="mx_AccessSecretStorageDialog_recoveryKeyEntry_entryControlSeparatorText">
                                 {_t("%(securityKey)s or %(recoveryFile)s", {
                                     recoveryFile: "",
                                     securityKey: "",
                                 })}
                             </span>
-                            <div>
+                             <div>
                                 <input
                                     type="file"
                                     className="mx_AccessSecretStorageDialog_recoveryKeyEntry_fileInput"
@@ -424,7 +426,8 @@ export default class AccessSecretStorageDialog extends React.PureComponent<IProp
                                 <AccessibleButton kind="primary" onClick={this.onRecoveryKeyFileUploadClick}>
                                     {_t("Upload")}
                                 </AccessibleButton>
-                            </div>
+                            :end TCHAP
+                            </div> */}
                         </div>
                         {recoveryKeyFeedback}
                         <DialogButtons
