diff --git a/node_modules/matrix-react-sdk/src/components/structures/RoomSearchView.tsx b/node_modules/matrix-react-sdk/src/components/structures/RoomSearchView.tsx
index bfa9930..245fc75 100644
--- a/node_modules/matrix-react-sdk/src/components/structures/RoomSearchView.tsx
+++ b/node_modules/matrix-react-sdk/src/components/structures/RoomSearchView.tsx
@@ -36,6 +36,7 @@ import ResizeNotifier from "../../utils/ResizeNotifier";
 import MatrixClientContext from "../../contexts/MatrixClientContext";
 import { RoomPermalinkCreator } from "../../utils/permalinks/Permalinks";
 import RoomContext from "../../contexts/RoomContext";
+import Tchapi18nUtils from "../../../../../src/tchap/i18n/Tchapi18nUtils";
 
 const DEBUG = false;
 let debuglog = function (msg: string): void {};
@@ -136,7 +137,8 @@ export const RoomSearchView = forwardRef<ScrollPanel, Props>(
                             logger.error("Search failed", error);
                             Modal.createDialog(ErrorDialog, {
                                 title: _t("error_dialog|search_failed|title"),
-                                description: error?.message ?? _t("error_dialog|search_failed|server_unavailable"),
+                                // :TCHAP: description: error?.message ?? _t("error_dialog|search_failed|server_unavailable"),
+                                description: error?.message ?? Tchapi18nUtils.getServerDownMessage(),
                             });
                             return false;
                         },
diff --git a/node_modules/matrix-react-sdk/src/components/structures/RoomStatusBar.tsx b/node_modules/matrix-react-sdk/src/components/structures/RoomStatusBar.tsx
index 657823f..46a424b 100644
--- a/node_modules/matrix-react-sdk/src/components/structures/RoomStatusBar.tsx
+++ b/node_modules/matrix-react-sdk/src/components/structures/RoomStatusBar.tsx
@@ -29,6 +29,7 @@ import InlineSpinner from "../views/elements/InlineSpinner";
 import MatrixClientContext from "../../contexts/MatrixClientContext";
 import { RoomStatusBarUnsentMessages } from "./RoomStatusBarUnsentMessages";
 import ExternalLink from "../views/elements/ExternalLink";
+import Tchapi18nUtils from '../../../../../src/tchap/i18n/Tchapi18nUtils'; // :TCHAP:
 
 const STATUS_BAR_HIDDEN = 0;
 const STATUS_BAR_EXPANDED = 1;
@@ -270,7 +271,9 @@ export default class RoomStatusBar extends React.PureComponent<IProps, IState> {
                             <WarningIcon width="24" height="24" />
                             <div>
                                 <div className="mx_RoomStatusBar_connectionLostBar_title">
-                                    {_t("room|status_bar|server_connectivity_lost_title")}
+                                    {/* :TCHAP: _t("room|status_bar|server_connectivity_lost_title") */}
+                                    {Tchapi18nUtils.getServerDownMessage()}
+                                    {/* end :TCHAP: */}
                                 </div>
                                 <div className="mx_RoomStatusBar_connectionLostBar_desc">
                                     {_t("room|status_bar|server_connectivity_lost_description")}
