diff --git a/node_modules/matrix-react-sdk/src/components/views/rooms/MemberList.tsx b/node_modules/matrix-react-sdk/src/components/views/rooms/MemberList.tsx
index 44dba2e..5232af5 100644
--- a/node_modules/matrix-react-sdk/src/components/views/rooms/MemberList.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/rooms/MemberList.tsx
@@ -39,6 +39,7 @@ import TruncatedList from "../elements/TruncatedList";
 import Spinner from "../elements/Spinner";
 import SearchBox from "../../structures/SearchBox";
 import AccessibleButton, { ButtonEvent } from "../elements/AccessibleButton";
+import AccessibleTooltipButton from "../elements/AccessibleTooltipButton"; // TCHAP
 import EntityTile from "./EntityTile";
 import MemberTile from "./MemberTile";
 import BaseAvatar from "../avatars/BaseAvatar";
@@ -46,6 +47,7 @@ import { shouldShowComponent } from "../../../customisations/helpers/UIComponent
 import { UIComponent } from "../../../settings/UIFeature";
 import PosthogTrackers from "../../../PosthogTrackers";
 import { SDKContext } from "../../../contexts/SDKContext";
+import { FileDownloader } from "../../../utils/FileDownloader";
 
 const INITIAL_LOAD_NUM_MEMBERS = 30;
 const INITIAL_LOAD_NUM_INVITED = 5;
@@ -73,6 +75,7 @@ export default class MemberList extends React.Component<IProps, IState> {
 
     public static contextType = SDKContext;
     public context!: React.ContextType<typeof SDKContext>;
+    private downloader = new FileDownloader(); // TCHAP
 
     public constructor(props: IProps, context: React.ContextType<typeof SDKContext>) {
         super(props);
@@ -368,6 +371,21 @@ export default class MemberList extends React.Component<IProps, IState> {
             );
         }
 
+        /* TCHAP */
+        let downloadMembersButton;
+        if (room?.getMyMembership() === "join" && !room.isSpaceRoom() && this.getChildCountInvited() > 0) {
+            downloadMembersButton = (
+                <AccessibleTooltipButton
+                    className="mx_MemberList_download mx_AccessibleButton mx_AccessibleButton_hasKind mx_AccessibleButton_kind_secondary"
+                    onClick={this.onDownloadButtonClick}
+                    tooltip={_t("Download the list of all this room's members, in a text file. Useful for adding them all to another room.")}
+                >
+                    <span>{_t("Export room members")}</span>
+                </AccessibleTooltipButton>
+            );
+        }
+        /* end TCHAP */
+
         let invitedHeader;
         let invitedSection;
         if (this.getChildCountInvited() > 0) {
@@ -409,6 +427,7 @@ export default class MemberList extends React.Component<IProps, IState> {
                     <React.Fragment>
                         {scopeHeader}
                         {inviteButton}
+                        {downloadMembersButton /* TCHAP */}
                     </React.Fragment>
                 }
                 footer={footer}
@@ -443,4 +462,24 @@ export default class MemberList extends React.Component<IProps, IState> {
             roomId: this.props.roomId,
         });
     };
+
+    /* TCHAP */
+    private onDownloadButtonClick = (ev: ButtonEvent): void => {
+        const members = this.state.filteredJoinedMembers.map(roomMember => roomMember.userId);
+        const blob = new Blob(members, { type : 'plain/text' })
+
+        const cli = MatrixClientPeg.get();
+        const room = cli.getRoom(this.props.roomId);
+        const filename = _t('members_of_%(roomName)s.txt', {
+            roomName: room.normalizedName,
+        });
+
+        this.downloader.download({
+            blob: blob,
+            name: filename,
+        });
+
+        return;
+    };
+    /* end TCHAP */
 }
