diff --git a/node_modules/matrix-react-sdk/src/hooks/useUserOnboardingTasks.ts b/node_modules/matrix-react-sdk/src/hooks/useUserOnboardingTasks.ts
index 43e819c..aa75cef 100644
--- a/node_modules/matrix-react-sdk/src/hooks/useUserOnboardingTasks.ts
+++ b/node_modules/matrix-react-sdk/src/hooks/useUserOnboardingTasks.ts
@@ -29,6 +29,7 @@ import SdkConfig from "../SdkConfig";
 import { UseCase } from "../settings/enums/UseCase";
 import { useSettingValue } from "./useSettings";
 import { UserOnboardingContext } from "./useUserOnboardingContext";
+import { accessSecretStorage } from "../SecurityManager";
 
 interface UserOnboardingTask {
     id: string;
@@ -54,12 +55,28 @@ const onClickStartDm = (ev: ButtonEvent): void => {
 };
 
 const tasks: UserOnboardingTask[] = [
+    /** :TCHAP: delete the first task, we have too many
     {
         id: "create-account",
         title: _t("Create account"),
         description: _t("onboarding|you_made_it"),
         completed: () => true,
     },
+    */
+    { // :TCHAP:
+        id: "setup-secure-backup",
+        title: _t("Sauvegarder vos messages automatiquement"),
+        description: _t("Vous pourrez les récupérez sur n'importe quel appareil à l'aide du Code de Récupération."),
+        completed: (ctx: UserOnboardingContext) => {
+
+        }, // todo
+        action: {
+            label: _t("Activer la sauvegarde automatique"),
+            onClick: (ev: ButtonEvent) => {
+                accessSecretStorage();
+            },
+        },
+    }, // end :TCHAP:
     {
         id: "find-friends",
         title: _t("onboarding|find_friends"),
