diff --git a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/HelpUserSettingsTab.tsx b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/HelpUserSettingsTab.tsx
index 7bb06f1..d2ee4d8 100644
--- a/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/HelpUserSettingsTab.tsx
+++ b/node_modules/matrix-react-sdk/src/components/views/settings/tabs/user/HelpUserSettingsTab.tsx
@@ -188,7 +188,9 @@ export default class HelpUserSettingsTab extends React.Component<IProps, IState>
             },
             {
                 'a': (sub) => <a
-                    href="https://element.io/help"
+                    // :TCHAP: fix url
+                    href="https://tchap.beta.gouv.fr/faq"
+                    // end :TCHAP:
                     rel="noreferrer noopener"
                     target="_blank"
                 >
@@ -267,19 +269,64 @@ export default class HelpUserSettingsTab extends React.Component<IProps, IState>
 
         const { appVersion, olmVersion } = this.getVersionInfo();
 
+        // :TCHAP: add Contact us section
+        const supportEmail = "support@tchap.gouv.fr";
+        const contactSection = (
+            <div>
+                <span className='mx_SettingsTab_subheading'>{ _t("Contact us") }</span>
+                <div className='mx_SettingsTab_subsectionText'>
+                    {
+                        _t(
+                            "If you have any difficulties with using Tchap, please contact us by email at <b>%(supportEmail)s</b>",
+                            { supportEmail: supportEmail },
+                            {
+                                "b": (sub) => <b>{ sub }</b>,
+                            }
+                        )
+                    }
+                </div>
+            </div>
+        );
+
+        const knownIssuesUrl = "https://github.com/tchapgouv/tchap-web-v4/wiki/Nouveau-Tchap-Web";
+        const knownIssuesSection = (
+            <div>
+                <span className='mx_SettingsTab_subheading'>{ _t("Known issues") }</span>
+                <div className='mx_SettingsTab_subsectionText'>
+                    {
+                        _t(
+                            "This version of Tchap Web is currently in development. There are known issues that will be solved soon." +
+                            " You can consult the list of known issues by <a>clicking here</a>.",
+                            {},
+                            {
+                                a: sub => <a href={knownIssuesUrl}
+                                    rel="noreferrer noopener"
+                                    target="_blank"
+                                >{ sub }</a>,
+                            }
+                        )
+                    }
+                </div>
+            </div>
+        );
+        // end :TCHAP:
+
         return (
             <div className="mx_SettingsTab mx_HelpUserSettingsTab">
                 <div className="mx_SettingsTab_heading">{ _t("Help & About") }</div>
-                { bugReportingSection }
+                { /* :TCHAP: move to lower down the page bugReportingSection */ }
                 <div className='mx_SettingsTab_section'>
+                    { /* :TCHAP: added */ contactSection }
                     <span className='mx_SettingsTab_subheading'>{ _t("FAQ") }</span>
                     <div className='mx_SettingsTab_subsectionText'>
                         { faqText }
                     </div>
-                    <AccessibleButton kind="primary" onClick={this.onKeyboardShortcutsClicked}>
+                    { /* :TCHAP: removed this shortcut, not that important <AccessibleButton kind="primary" onClick={this.onKeyboardShortcutsClicked}>
                         { _t("Keyboard Shortcuts") }
-                    </AccessibleButton>
+                      </AccessibleButton> */}
+                    { /* :TCHAP: added */ knownIssuesSection }
                 </div>
+                { /* :TCHAP: moved from higher up the page */ bugReportingSection }
                 <div className='mx_SettingsTab_section'>
                     <span className='mx_SettingsTab_subheading'>{ _t("Versions") }</span>
                     <div className='mx_SettingsTab_subsectionText'>
